<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.29">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Benjamin Fair">
<meta name="dcterms.date" content="2025-08-05">

<title>20250716_Explore Differential Expression and Splicing Results – 20250624_PanExperiment_NMD_CandidateJuncs</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-0815c480559380816a4d1ea211a47e91.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-c6aa55ccafe579b599e1c08b0f990df4.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">20250624_PanExperiment_NMD_CandidateJuncs</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./about.html"> 
<span class="menu-text">About</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction">Introduction</a></li>
  <li><a href="#code" id="toc-code" class="nav-link" data-scroll-target="#code">Code</a>
  <ul class="collapse">
  <li><a href="#summary-plots" id="toc-summary-plots" class="nav-link" data-scroll-target="#summary-plots">Summary plots</a></li>
  <li><a href="#check-example-of-known-nmd-targets" id="toc-check-example-of-known-nmd-targets" class="nav-link" data-scroll-target="#check-example-of-known-nmd-targets">Check example of known NMD targets</a></li>
  <li><a href="#mbd5" id="toc-mbd5" class="nav-link" data-scroll-target="#mbd5">MBD5</a></li>
  <li><a href="#utr-splicing" id="toc-utr-splicing" class="nav-link" data-scroll-target="#utr-splicing">5’ UTR splicing</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">20250716_Explore Differential Expression and Splicing Results</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Benjamin Fair </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">August 5, 2025</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<section id="introduction" class="level2">
<h2 class="anchored" data-anchor-id="introduction">Introduction</h2>
<p>I have done my standard differential expression analysis with edgeR, and differential splicing analysis with leafcutter, for a bunch of contrasts that in some sense compare NMD enriched datasets to control. Here I will explore the results.</p>
</section>
<section id="code" class="level2">
<h2 class="anchored" data-anchor-id="code">Code</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(valr)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="co"># theshold in percent</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>dPSI_threshold <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>diff.splicing.results <span class="ot">&lt;-</span> <span class="fu">Sys.glob</span>(<span class="st">"../code/rna_seq/SplicingAnalysis/differential_splicing_tidy/*/Results.tsv.gz"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">setNames</span>(<span class="fu">str_replace</span>(., <span class="st">"../code/rna_seq/SplicingAnalysis/differential_splicing_tidy/(.+?)/Results.tsv.gz"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">lapply</span>(fread) <span class="sc">%&gt;%</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"Contrast"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">deltapsi =</span> deltapsi <span class="sc">*</span> <span class="dv">100</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">IsSig =</span> p.adjust <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(deltapsi) <span class="sc">&gt;</span> dPSI_threshold)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>Gene_symbols <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"/project2/yangili1/bjf79/ReferenceGenomes/Human_UCSC.hg38_GencodeComprehensive46/GeneSymbols.tsv.gz"</span>, <span class="at">col_names=</span><span class="fu">c</span>(<span class="st">"Transcript"</span>, <span class="st">"Symbol"</span>, <span class="st">"HGNC_id"</span>))</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>Gene_names <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"/project2/yangili1/bjf79/ReferenceGenomes/Human_UCSC.hg38_GencodeComprehensive46/Reference.ColoredTranscripts.bed.gz"</span>, <span class="at">header=</span>F) <span class="sc">%&gt;%</span></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">inner_join</span>(</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a>        Gene_symbols,</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"V4"</span> <span class="ot">=</span> <span class="st">"Transcript"</span>)</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">distinct</span>(<span class="at">GeneID=</span>V13, Symbol)</span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a>DE.results <span class="ot">&lt;-</span> <span class="fu">Sys.glob</span>(<span class="st">"../code/rna_seq/differential_expression/*/results.tsv.gz"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">setNames</span>(<span class="fu">str_replace</span>(., <span class="st">"../code/rna_seq/differential_expression/(.+?)/results.tsv.gz"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">lapply</span>(fread) <span class="sc">%&gt;%</span></span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">"condition"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(Gene_names, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"GeneID"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="summary-plots" class="level3">
<h3 class="anchored" data-anchor-id="summary-plots">Summary plots</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>DE.results <span class="sc">%&gt;%</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">IsSig =</span> <span class="fu">case_when</span>(</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>        FDR <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(logFC) <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">~</span> <span class="st">"FDR&lt;1% &amp; |logFC|&gt;1"</span>,</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>        FDR <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(logFC) <span class="sc">&lt;=</span> <span class="dv">1</span> <span class="sc">~</span> <span class="st">"FDR&lt;1% &amp; |logFC|&lt;=1"</span>,</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>        <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"NS"</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>    )) <span class="sc">%&gt;%</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">IsSig =</span> <span class="fu">factor</span>(IsSig, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">"FDR&lt;1% &amp; |logFC|&gt;1"</span>, <span class="st">"FDR&lt;1% &amp; |logFC|&lt;=1"</span>, <span class="st">"NS"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">count</span>(condition, IsSig) <span class="sc">%&gt;%</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y=</span>condition, <span class="at">x=</span>n, <span class="at">fill=</span>IsSig)) <span class="sc">+</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>    <span class="co"># geom_label_repel(data = . %&gt;%</span></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>    <span class="co">#     filter(!IsSig == "NS"),</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>    <span class="co">#     aes(label=n), hjust=0, position = position_stack()) +</span></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"Number of genes"</span>, <span class="at">y=</span><span class="st">"Condition"</span>, <span class="at">fill=</span><span class="st">"Significance"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="20250716_Explore_DifferentialExpressionSplicingResults_files/figure-html/unnamed-chunk-2-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>from this cyclohexamide vs dmso is probably the best contrast… Let’s look at differential splicing</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>diff.splicing.results <span class="sc">%&gt;%</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(IsSig) <span class="sc">%&gt;%</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">distinct</span>(Contrast, cluster, <span class="at">.keep_all=</span>T) <span class="sc">%&gt;%</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">count</span>(Contrast) <span class="sc">%&gt;%</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(Contrast) <span class="sc">%&gt;%</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y=</span>Contrast, <span class="at">x=</span>n)) <span class="sc">+</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_col</span>(<span class="at">position=</span><span class="st">'stack'</span>) <span class="sc">+</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label=</span>n), <span class="at">hjust=</span><span class="sc">-</span><span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_discrete</span>(<span class="at">limits =</span> rev) <span class="sc">+</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_continuous</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.1</span>))) <span class="sc">+</span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"Number of leafcutter significant clusters"</span>, <span class="at">y=</span><span class="st">"Contrast"</span>, <span class="at">caption=</span><span class="fu">str_glue</span>(<span class="st">"FDR&lt;1% &amp; |dPSI|&gt;{dPSI_threshold}%"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="20250716_Explore_DifferentialExpressionSplicingResults_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Let’s also look at effect sizes of the unproductive vs productive</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>diff.splicing.results <span class="sc">%&gt;%</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(IsSig) <span class="sc">%&gt;%</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Contrast, cluster, Coding) <span class="sc">%&gt;%</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(<span class="fu">abs</span>(deltapsi))) <span class="sc">%&gt;%</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">distinct</span>(Contrast, cluster, Coding, <span class="at">.keep_all=</span>T) <span class="sc">%&gt;%</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Coding)) <span class="sc">%&gt;%</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>logef, <span class="at">color=</span>Coding)) <span class="sc">+</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">0</span>, <span class="at">color=</span><span class="st">'gray'</span>) <span class="sc">+</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stat_ecdf</span>() <span class="sc">+</span> </span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>( <span class="at">data =</span> . <span class="sc">%&gt;%</span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>        <span class="fu">group_by</span>(Contrast, Coding) <span class="sc">%&gt;%</span></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>        <span class="fu">summarise</span>(<span class="at">median=</span><span class="fu">median</span>(logef, <span class="at">na.rm=</span>T)),</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(<span class="at">xintercept=</span>median, <span class="at">color=</span>Coding), <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>Contrast) <span class="sc">+</span></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_cartesian</span>(<span class="at">xlim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>, <span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="20250716_Explore_DifferentialExpressionSplicingResults_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Ok, so from here, I don’t think mine/Sara’s polysome profilnig libraries are useful. All the other contrasts seem more reasonable in the sense that of the differential spliced unproductive junctions, they are generally up-regulated in the NMD enriched conditions, and the productive junctions are down-regulated, and mine as well filter out the Sara samples to avoid too many points/colors on plots.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>diff.splicing.results <span class="ot">&lt;-</span> diff.splicing.results <span class="sc">%&gt;%</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(Contrast, <span class="st">"Sara"</span>))</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>DE.results <span class="ot">&lt;-</span> DE.results <span class="sc">%&gt;%</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(condition, <span class="st">"Sara"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="check-example-of-known-nmd-targets" class="level3">
<h3 class="anchored" data-anchor-id="check-example-of-known-nmd-targets">Check example of known NMD targets</h3>
<p>Let’s look at a few junctions of intersest in known NMD targets…</p>
<p>First let’s look at SRSF6, which is a known NMD target.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>diff.splicing.results <span class="sc">%&gt;%</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Gene_name<span class="sc">==</span><span class="st">"SRSF6"</span> <span class="sc">&amp;</span> <span class="sc">!</span>Coding)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                                       Contrast
                                                         &lt;char&gt;
 1:                           Colombo.SM67_and_SMG7_vs_Scramble
 2:                           Colombo.SM67_and_SMG7_vs_Scramble
 3:                                    Colombo.UPF1_vs_Scramble
 4:                                    Colombo.UPF1_vs_Scramble
 5:                                    Colombo.UPF1_vs_Scramble
 6:                               Darman_Cycloxhexamide_vs_DMSO
 7:                               Darman_Cycloxhexamide_vs_DMSO
 8:  DylanUnpublishedFractionation.JurkatNuclear_vs_Cytoplasmic
 9:  DylanUnpublishedFractionation.JurkatNuclear_vs_Cytoplasmic
10:  DylanUnpublishedFractionation.JurkatNuclear_vs_Cytoplasmic
11:     DylanUnpublishedFractionation.LCLNuclear_vs_Cytoplasmic
12:     DylanUnpublishedFractionation.LCLNuclear_vs_Cytoplasmic
13:     DylanUnpublishedFractionation.LCLNuclear_vs_Cytoplasmic
14: DylanUnpublishedFractionation.SH-SY5YNuclear_vs_Cytoplasmic
15: DylanUnpublishedFractionation.SH-SY5YNuclear_vs_Cytoplasmic
16:                        FairAndGEUVADIS_naRNA_vs_SteadyState
17:                        FairAndGEUVADIS_naRNA_vs_SteadyState
18:                        FairAndGEUVADIS_naRNA_vs_SteadyState
19:                        Floor_LightPolysome_vs_HeavyPolysome
20:                        Floor_LightPolysome_vs_HeavyPolysome
21:                    Ritter.ESCLightPolysome_vs_HeavyPolysome
22:                    Ritter.ESCLightPolysome_vs_HeavyPolysome
23:                    Ritter.ESCLightPolysome_vs_HeavyPolysome
24:                    Ritter.NPCLightPolysome_vs_HeavyPolysome
25:                    Ritter.NPCLightPolysome_vs_HeavyPolysome
26:                    Ritter.NPCLightPolysome_vs_HeavyPolysome
                                                       Contrast
                                 intron  chrom    start      end
                                 &lt;char&gt; &lt;char&gt;    &lt;int&gt;    &lt;int&gt;
 1: chr20:43458509:43459153:clu_53323_+  chr20 43458509 43459153
 2: chr20:43459420:43459771:clu_53323_+  chr20 43459420 43459771
 3: chr20:43458509:43459153:clu_53323_+  chr20 43458509 43459153
 4: chr20:43458729:43459153:clu_53323_+  chr20 43458729 43459153
 5: chr20:43459420:43459771:clu_53323_+  chr20 43459420 43459771
 6: chr20:43458509:43459153:clu_53323_+  chr20 43458509 43459153
 7: chr20:43459420:43459771:clu_53323_+  chr20 43459420 43459771
 8: chr20:43458509:43459153:clu_53323_+  chr20 43458509 43459153
 9: chr20:43458729:43459153:clu_53323_+  chr20 43458729 43459153
10: chr20:43459420:43459771:clu_53323_+  chr20 43459420 43459771
11: chr20:43458509:43459153:clu_53323_+  chr20 43458509 43459153
12: chr20:43458729:43459153:clu_53323_+  chr20 43458729 43459153
13: chr20:43459420:43459771:clu_53323_+  chr20 43459420 43459771
14: chr20:43458509:43459153:clu_53323_+  chr20 43458509 43459153
15: chr20:43459420:43459771:clu_53323_+  chr20 43459420 43459771
16: chr20:43458509:43459153:clu_53323_+  chr20 43458509 43459153
17: chr20:43458729:43459153:clu_53323_+  chr20 43458729 43459153
18: chr20:43459420:43459771:clu_53323_+  chr20 43459420 43459771
19: chr20:43458509:43459153:clu_53323_+  chr20 43458509 43459153
20: chr20:43459420:43459771:clu_53323_+  chr20 43459420 43459771
21: chr20:43458509:43459153:clu_53323_+  chr20 43458509 43459153
22: chr20:43458729:43459153:clu_53323_+  chr20 43458729 43459153
23: chr20:43459420:43459771:clu_53323_+  chr20 43459420 43459771
24: chr20:43458509:43459153:clu_53323_+  chr20 43458509 43459153
25: chr20:43458729:43459153:clu_53323_+  chr20 43458729 43459153
26: chr20:43459420:43459771:clu_53323_+  chr20 43459420 43459771
                                 intron  chrom    start      end
              cluster       logef      Control NMD_enriched     deltapsi
               &lt;char&gt;       &lt;num&gt;        &lt;num&gt;        &lt;num&gt;        &lt;num&gt;
 1: chr20:clu_53323_+  1.05206545 0.0161922606 0.2453837535 22.919149291
 2: chr20:clu_53323_+  0.24074416 0.0512307019 0.3449183243 29.368762243
 3: chr20:clu_53323_+  0.46177981 0.0156110615 0.0832650872  6.765402571
 4: chr20:clu_53323_+  3.37877016 0.0001048497 0.0103378590  1.023300927
 5: chr20:clu_53323_+ -0.32040107 0.0513133390 0.1251883579  7.387501888
 6: chr20:clu_53323_+  0.46984437 0.0237465739 0.0707157751  4.696920120
 7: chr20:clu_53323_+  0.26200835 0.0972198706 0.2351843117 13.796444110
 8: chr20:clu_53323_+ -0.09305403 0.0130306847 0.0368461784  2.381549363
 9: chr20:clu_53323_+ -1.70055505 0.0004663993 0.0002642739 -0.020212540
10: chr20:clu_53323_+  0.26586046 0.0269501093 0.1091089366  8.215882728
11: chr20:clu_53323_+  0.02996170 0.0078785670 0.0304406957  2.256212871
12: chr20:clu_53323_+ -1.65009747 0.0026140760 0.0018822839 -0.073179209
13: chr20:clu_53323_+  0.59685160 0.0400033187 0.2724586007 23.245528198
14: chr20:clu_53323_+ -1.73479925 0.0477505890 0.0251624315 -2.258815750
15: chr20:clu_53323_+ -0.78669334 0.2546476482 0.3463145270  9.166687876
16: chr20:clu_53323_+  0.20281255 0.0256981574 0.0310789590  0.538080166
17: chr20:clu_53323_+ -0.11980759 0.0003037903 0.0002660885 -0.003770171
18: chr20:clu_53323_+  0.28691998 0.1182533395 0.1555626337  3.730929412
19: chr20:clu_53323_+  0.48096831 0.0460642384 0.0643428282  1.827858981
20: chr20:clu_53323_+  1.84131034 0.0958023275 0.5215564848 42.575415727
21: chr20:clu_53323_+ -0.01417687 0.0414342418 0.0960540993  5.461985745
22: chr20:clu_53323_+  2.33191681 0.0003859043 0.0093439760  0.895807165
23: chr20:clu_53323_+  0.28636818 0.1156474218 0.3620908219 24.644340009
24: chr20:clu_53323_+  0.16590115 0.0947353722 0.1218323645  2.709699225
25: chr20:clu_53323_+  1.25729417 0.0032633497 0.0124997145  0.923636475
26: chr20:clu_53323_+  0.51797121 0.2563004974 0.4687076483 21.240715094
              cluster       logef      Control NMD_enriched     deltapsi
     status     loglr    df            p     p.adjust strand
     &lt;char&gt;     &lt;num&gt; &lt;int&gt;        &lt;num&gt;        &lt;num&gt; &lt;char&gt;
 1: Success 57.715717     6 1.482475e-22 1.416475e-20      +
 2: Success 57.715717     6 1.482475e-22 1.416475e-20      +
 3: Success 33.255349     6 2.119251e-12 2.782228e-10      +
 4: Success 33.255349     6 2.119251e-12 2.782228e-10      +
 5: Success 33.255349     6 2.119251e-12 2.782228e-10      +
 6: Success 30.815444     6 2.097478e-11 6.007507e-10      +
 7: Success 30.815444     6 2.097478e-11 6.007507e-10      +
 8: Success 42.693549     8 4.001478e-15 3.988212e-13      +
 9: Success 42.693549     8 4.001478e-15 3.988212e-13      +
10: Success 42.693549     8 4.001478e-15 3.988212e-13      +
11: Success 20.172608     7 1.081055e-06 1.412099e-05      +
12: Success 20.172608     7 1.081055e-06 1.412099e-05      +
13: Success 20.172608     7 1.081055e-06 1.412099e-05      +
14: Success  8.475313     5 4.594393e-03 2.127349e-02      +
15: Success  8.475313     5 4.594393e-03 2.127349e-02      +
16: Success 26.926987     8 7.368617e-09 1.535400e-08      +
17: Success 26.926987     8 7.368617e-09 1.535400e-08      +
18: Success 26.926987     8 7.368617e-09 1.535400e-08      +
19: Success 16.263948     5 4.669469e-06 4.472329e-05      +
20: Success 16.263948     5 4.669469e-06 4.472329e-05      +
21: Success 65.386372     7 4.333776e-25 4.080603e-22      +
22: Success 65.386372     7 4.333776e-25 4.080603e-22      +
23: Success 65.386372     7 4.333776e-25 4.080603e-22      +
24: Success 15.432546     7 6.583855e-05 1.207907e-03      +
25: Success 15.432546     7 6.583855e-05 1.207907e-03      +
26: Success 15.432546     7 6.583855e-05 1.207907e-03      +
     status     loglr    df            p     p.adjust strand
                         name score splice_site acceptors_skipped exons_skipped
                       &lt;char&gt; &lt;int&gt;      &lt;char&gt;             &lt;int&gt;         &lt;int&gt;
 1: chr20_43458509_43459152_+  8759       GT-AG                 0             0
 2: chr20_43459420_43459770_+ 29930       GT-AG                 0             0
 3: chr20_43458509_43459152_+  8759       GT-AG                 0             0
 4: chr20_43458729_43459152_+   158       GT-AG                 0             0
 5: chr20_43459420_43459770_+ 29930       GT-AG                 0             0
 6: chr20_43458509_43459152_+  8759       GT-AG                 0             0
 7: chr20_43459420_43459770_+ 29930       GT-AG                 0             0
 8: chr20_43458509_43459152_+  8759       GT-AG                 0             0
 9: chr20_43458729_43459152_+   158       GT-AG                 0             0
10: chr20_43459420_43459770_+ 29930       GT-AG                 0             0
11: chr20_43458509_43459152_+  8759       GT-AG                 0             0
12: chr20_43458729_43459152_+   158       GT-AG                 0             0
13: chr20_43459420_43459770_+ 29930       GT-AG                 0             0
14: chr20_43458509_43459152_+  8759       GT-AG                 0             0
15: chr20_43459420_43459770_+ 29930       GT-AG                 0             0
16: chr20_43458509_43459152_+  8759       GT-AG                 0             0
17: chr20_43458729_43459152_+   158       GT-AG                 0             0
18: chr20_43459420_43459770_+ 29930       GT-AG                 0             0
19: chr20_43458509_43459152_+  8759       GT-AG                 0             0
20: chr20_43459420_43459770_+ 29930       GT-AG                 0             0
21: chr20_43458509_43459152_+  8759       GT-AG                 0             0
22: chr20_43458729_43459152_+   158       GT-AG                 0             0
23: chr20_43459420_43459770_+ 29930       GT-AG                 0             0
24: chr20_43458509_43459152_+  8759       GT-AG                 0             0
25: chr20_43458729_43459152_+   158       GT-AG                 0             0
26: chr20_43459420_43459770_+ 29930       GT-AG                 0             0
                         name score splice_site acceptors_skipped exons_skipped
    donors_skipped anchor known_donor known_acceptor known_junction
             &lt;int&gt; &lt;char&gt;       &lt;int&gt;          &lt;int&gt;          &lt;int&gt;
 1:              0      D           1              0              0
 2:              0      A           0              1              0
 3:              0      D           1              0              0
 4:              0      N           0              0              0
 5:              0      A           0              1              0
 6:              0      D           1              0              0
 7:              0      A           0              1              0
 8:              0      D           1              0              0
 9:              0      N           0              0              0
10:              0      A           0              1              0
11:              0      D           1              0              0
12:              0      N           0              0              0
13:              0      A           0              1              0
14:              0      D           1              0              0
15:              0      A           0              1              0
16:              0      D           1              0              0
17:              0      N           0              0              0
18:              0      A           0              1              0
19:              0      D           1              0              0
20:              0      A           0              1              0
21:              0      D           1              0              0
22:              0      N           0              0              0
23:              0      A           0              1              0
24:              0      D           1              0              0
25:              0      N           0              0              0
26:              0      A           0              1              0
    donors_skipped anchor known_donor known_acceptor known_junction
               gene_names                             gene_ids
                   &lt;char&gt;                               &lt;char&gt;
 1: ENSG00000288000,SRSF6 ENSG00000288000.1,ENSG00000124193.16
 2: ENSG00000288000,SRSF6 ENSG00000288000.1,ENSG00000124193.16
 3: ENSG00000288000,SRSF6 ENSG00000288000.1,ENSG00000124193.16
 4:                                                           
 5: ENSG00000288000,SRSF6 ENSG00000288000.1,ENSG00000124193.16
 6: ENSG00000288000,SRSF6 ENSG00000288000.1,ENSG00000124193.16
 7: ENSG00000288000,SRSF6 ENSG00000288000.1,ENSG00000124193.16
 8: ENSG00000288000,SRSF6 ENSG00000288000.1,ENSG00000124193.16
 9:                                                           
10: ENSG00000288000,SRSF6 ENSG00000288000.1,ENSG00000124193.16
11: ENSG00000288000,SRSF6 ENSG00000288000.1,ENSG00000124193.16
12:                                                           
13: ENSG00000288000,SRSF6 ENSG00000288000.1,ENSG00000124193.16
14: ENSG00000288000,SRSF6 ENSG00000288000.1,ENSG00000124193.16
15: ENSG00000288000,SRSF6 ENSG00000288000.1,ENSG00000124193.16
16: ENSG00000288000,SRSF6 ENSG00000288000.1,ENSG00000124193.16
17:                                                           
18: ENSG00000288000,SRSF6 ENSG00000288000.1,ENSG00000124193.16
19: ENSG00000288000,SRSF6 ENSG00000288000.1,ENSG00000124193.16
20: ENSG00000288000,SRSF6 ENSG00000288000.1,ENSG00000124193.16
21: ENSG00000288000,SRSF6 ENSG00000288000.1,ENSG00000124193.16
22:                                                           
23: ENSG00000288000,SRSF6 ENSG00000288000.1,ENSG00000124193.16
24: ENSG00000288000,SRSF6 ENSG00000288000.1,ENSG00000124193.16
25:                                                           
26: ENSG00000288000,SRSF6 ENSG00000288000.1,ENSG00000124193.16
               gene_names                             gene_ids
                                                                                  transcripts
                                                                                       &lt;char&gt;
 1: ENST00000244020.5,ENST00000657241.1,ENST00000662078.1,ENST00000668808.1,ENST00000670741.1
 2: ENST00000244020.5,ENST00000657241.1,ENST00000662078.1,ENST00000668808.1,ENST00000670741.1
 3: ENST00000244020.5,ENST00000657241.1,ENST00000662078.1,ENST00000668808.1,ENST00000670741.1
 4:                                                                                          
 5: ENST00000244020.5,ENST00000657241.1,ENST00000662078.1,ENST00000668808.1,ENST00000670741.1
 6: ENST00000244020.5,ENST00000657241.1,ENST00000662078.1,ENST00000668808.1,ENST00000670741.1
 7: ENST00000244020.5,ENST00000657241.1,ENST00000662078.1,ENST00000668808.1,ENST00000670741.1
 8: ENST00000244020.5,ENST00000657241.1,ENST00000662078.1,ENST00000668808.1,ENST00000670741.1
 9:                                                                                          
10: ENST00000244020.5,ENST00000657241.1,ENST00000662078.1,ENST00000668808.1,ENST00000670741.1
11: ENST00000244020.5,ENST00000657241.1,ENST00000662078.1,ENST00000668808.1,ENST00000670741.1
12:                                                                                          
13: ENST00000244020.5,ENST00000657241.1,ENST00000662078.1,ENST00000668808.1,ENST00000670741.1
14: ENST00000244020.5,ENST00000657241.1,ENST00000662078.1,ENST00000668808.1,ENST00000670741.1
15: ENST00000244020.5,ENST00000657241.1,ENST00000662078.1,ENST00000668808.1,ENST00000670741.1
16: ENST00000244020.5,ENST00000657241.1,ENST00000662078.1,ENST00000668808.1,ENST00000670741.1
17:                                                                                          
18: ENST00000244020.5,ENST00000657241.1,ENST00000662078.1,ENST00000668808.1,ENST00000670741.1
19: ENST00000244020.5,ENST00000657241.1,ENST00000662078.1,ENST00000668808.1,ENST00000670741.1
20: ENST00000244020.5,ENST00000657241.1,ENST00000662078.1,ENST00000668808.1,ENST00000670741.1
21: ENST00000244020.5,ENST00000657241.1,ENST00000662078.1,ENST00000668808.1,ENST00000670741.1
22:                                                                                          
23: ENST00000244020.5,ENST00000657241.1,ENST00000662078.1,ENST00000668808.1,ENST00000670741.1
24: ENST00000244020.5,ENST00000657241.1,ENST00000662078.1,ENST00000668808.1,ENST00000670741.1
25:                                                                                          
26: ENST00000244020.5,ENST00000657241.1,ENST00000662078.1,ENST00000668808.1,ENST00000670741.1
                                                                                  transcripts
        5ss_seq    3ss_seq 5ss_score 3ss_score            Intron_coord
         &lt;char&gt;     &lt;char&gt;     &lt;num&gt;     &lt;num&gt;                  &lt;char&gt;
 1: CGCAGTGAGTC AATTTGAAAG  4.984518 0.6885456 chr20:43458509-43459153
 2: TGATGTAAGTA TGTGATAAAG  8.411946 1.4627157 chr20:43459420-43459771
 3: CGCAGTGAGTC AATTTGAAAG  4.984518 0.6885456 chr20:43458509-43459153
 4: GGAGGTGGGTG AATTTGAAAG  8.606801 0.6885456 chr20:43458729-43459153
 5: TGATGTAAGTA TGTGATAAAG  8.411946 1.4627157 chr20:43459420-43459771
 6: CGCAGTGAGTC AATTTGAAAG  4.984518 0.6885456 chr20:43458509-43459153
 7: TGATGTAAGTA TGTGATAAAG  8.411946 1.4627157 chr20:43459420-43459771
 8: CGCAGTGAGTC AATTTGAAAG  4.984518 0.6885456 chr20:43458509-43459153
 9: GGAGGTGGGTG AATTTGAAAG  8.606801 0.6885456 chr20:43458729-43459153
10: TGATGTAAGTA TGTGATAAAG  8.411946 1.4627157 chr20:43459420-43459771
11: CGCAGTGAGTC AATTTGAAAG  4.984518 0.6885456 chr20:43458509-43459153
12: GGAGGTGGGTG AATTTGAAAG  8.606801 0.6885456 chr20:43458729-43459153
13: TGATGTAAGTA TGTGATAAAG  8.411946 1.4627157 chr20:43459420-43459771
14: CGCAGTGAGTC AATTTGAAAG  4.984518 0.6885456 chr20:43458509-43459153
15: TGATGTAAGTA TGTGATAAAG  8.411946 1.4627157 chr20:43459420-43459771
16: CGCAGTGAGTC AATTTGAAAG  4.984518 0.6885456 chr20:43458509-43459153
17: GGAGGTGGGTG AATTTGAAAG  8.606801 0.6885456 chr20:43458729-43459153
18: TGATGTAAGTA TGTGATAAAG  8.411946 1.4627157 chr20:43459420-43459771
19: CGCAGTGAGTC AATTTGAAAG  4.984518 0.6885456 chr20:43458509-43459153
20: TGATGTAAGTA TGTGATAAAG  8.411946 1.4627157 chr20:43459420-43459771
21: CGCAGTGAGTC AATTTGAAAG  4.984518 0.6885456 chr20:43458509-43459153
22: GGAGGTGGGTG AATTTGAAAG  8.606801 0.6885456 chr20:43458729-43459153
23: TGATGTAAGTA TGTGATAAAG  8.411946 1.4627157 chr20:43459420-43459771
24: CGCAGTGAGTC AATTTGAAAG  4.984518 0.6885456 chr20:43458509-43459153
25: GGAGGTGGGTG AATTTGAAAG  8.606801 0.6885456 chr20:43458729-43459153
26: TGATGTAAGTA TGTGATAAAG  8.411946 1.4627157 chr20:43459420-43459771
        5ss_seq    3ss_seq 5ss_score 3ss_score            Intron_coord
    Gene_name  Annot Coding    UTR  IsSig
       &lt;char&gt; &lt;lgcl&gt; &lt;lgcl&gt; &lt;lgcl&gt; &lt;lgcl&gt;
 1:     SRSF6   TRUE  FALSE  FALSE   TRUE
 2:     SRSF6   TRUE  FALSE  FALSE   TRUE
 3:     SRSF6   TRUE  FALSE  FALSE   TRUE
 4:     SRSF6   TRUE  FALSE  FALSE   TRUE
 5:     SRSF6   TRUE  FALSE  FALSE   TRUE
 6:     SRSF6   TRUE  FALSE  FALSE   TRUE
 7:     SRSF6   TRUE  FALSE  FALSE   TRUE
 8:     SRSF6   TRUE  FALSE  FALSE   TRUE
 9:     SRSF6   TRUE  FALSE  FALSE  FALSE
10:     SRSF6   TRUE  FALSE  FALSE   TRUE
11:     SRSF6   TRUE  FALSE  FALSE   TRUE
12:     SRSF6   TRUE  FALSE  FALSE  FALSE
13:     SRSF6   TRUE  FALSE  FALSE   TRUE
14:     SRSF6   TRUE  FALSE  FALSE  FALSE
15:     SRSF6   TRUE  FALSE  FALSE  FALSE
16:     SRSF6   TRUE  FALSE  FALSE  FALSE
17:     SRSF6   TRUE  FALSE  FALSE  FALSE
18:     SRSF6   TRUE  FALSE  FALSE   TRUE
19:     SRSF6   TRUE  FALSE  FALSE   TRUE
20:     SRSF6   TRUE  FALSE  FALSE   TRUE
21:     SRSF6   TRUE  FALSE  FALSE   TRUE
22:     SRSF6   TRUE  FALSE  FALSE  FALSE
23:     SRSF6   TRUE  FALSE  FALSE   TRUE
24:     SRSF6   TRUE  FALSE  FALSE   TRUE
25:     SRSF6   TRUE  FALSE  FALSE  FALSE
26:     SRSF6   TRUE  FALSE  FALSE   TRUE
    Gene_name  Annot Coding    UTR  IsSig</code></pre>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>diff.splicing.results <span class="sc">%&gt;%</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Gene_name<span class="sc">==</span><span class="st">"SRSF6"</span> <span class="sc">&amp;</span> <span class="sc">!</span>Coding) <span class="sc">%&gt;%</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">color=</span>Contrast, <span class="at">x=</span>deltapsi, <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(p.adjust))) <span class="sc">+</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.01</span>), <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>intron) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="20250716_Explore_DifferentialExpressionSplicingResults_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>diff.splicing.results <span class="sc">%&gt;%</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Gene_name<span class="sc">==</span><span class="st">"SRSF6"</span> <span class="sc">&amp;</span> <span class="sc">!</span>Coding) <span class="sc">%&gt;%</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(intron <span class="sc">==</span> <span class="st">"chr20:43459420:43459771:clu_53323_+"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                                       Contrast
                                                         &lt;char&gt;
 1:                           Colombo.SM67_and_SMG7_vs_Scramble
 2:                                    Colombo.UPF1_vs_Scramble
 3:                               Darman_Cycloxhexamide_vs_DMSO
 4:  DylanUnpublishedFractionation.JurkatNuclear_vs_Cytoplasmic
 5:     DylanUnpublishedFractionation.LCLNuclear_vs_Cytoplasmic
 6: DylanUnpublishedFractionation.SH-SY5YNuclear_vs_Cytoplasmic
 7:                        FairAndGEUVADIS_naRNA_vs_SteadyState
 8:                        Floor_LightPolysome_vs_HeavyPolysome
 9:                    Ritter.ESCLightPolysome_vs_HeavyPolysome
10:                    Ritter.NPCLightPolysome_vs_HeavyPolysome
                                 intron  chrom    start      end
                                 &lt;char&gt; &lt;char&gt;    &lt;int&gt;    &lt;int&gt;
 1: chr20:43459420:43459771:clu_53323_+  chr20 43459420 43459771
 2: chr20:43459420:43459771:clu_53323_+  chr20 43459420 43459771
 3: chr20:43459420:43459771:clu_53323_+  chr20 43459420 43459771
 4: chr20:43459420:43459771:clu_53323_+  chr20 43459420 43459771
 5: chr20:43459420:43459771:clu_53323_+  chr20 43459420 43459771
 6: chr20:43459420:43459771:clu_53323_+  chr20 43459420 43459771
 7: chr20:43459420:43459771:clu_53323_+  chr20 43459420 43459771
 8: chr20:43459420:43459771:clu_53323_+  chr20 43459420 43459771
 9: chr20:43459420:43459771:clu_53323_+  chr20 43459420 43459771
10: chr20:43459420:43459771:clu_53323_+  chr20 43459420 43459771
              cluster      logef    Control NMD_enriched  deltapsi  status
               &lt;char&gt;      &lt;num&gt;      &lt;num&gt;        &lt;num&gt;     &lt;num&gt;  &lt;char&gt;
 1: chr20:clu_53323_+  0.2407442 0.05123070    0.3449183 29.368762 Success
 2: chr20:clu_53323_+ -0.3204011 0.05131334    0.1251884  7.387502 Success
 3: chr20:clu_53323_+  0.2620083 0.09721987    0.2351843 13.796444 Success
 4: chr20:clu_53323_+  0.2658605 0.02695011    0.1091089  8.215883 Success
 5: chr20:clu_53323_+  0.5968516 0.04000332    0.2724586 23.245528 Success
 6: chr20:clu_53323_+ -0.7866933 0.25464765    0.3463145  9.166688 Success
 7: chr20:clu_53323_+  0.2869200 0.11825334    0.1555626  3.730929 Success
 8: chr20:clu_53323_+  1.8413103 0.09580233    0.5215565 42.575416 Success
 9: chr20:clu_53323_+  0.2863682 0.11564742    0.3620908 24.644340 Success
10: chr20:clu_53323_+  0.5179712 0.25630050    0.4687076 21.240715 Success
        loglr    df            p     p.adjust strand                      name
        &lt;num&gt; &lt;int&gt;        &lt;num&gt;        &lt;num&gt; &lt;char&gt;                    &lt;char&gt;
 1: 57.715717     6 1.482475e-22 1.416475e-20      + chr20_43459420_43459770_+
 2: 33.255349     6 2.119251e-12 2.782228e-10      + chr20_43459420_43459770_+
 3: 30.815444     6 2.097478e-11 6.007507e-10      + chr20_43459420_43459770_+
 4: 42.693549     8 4.001478e-15 3.988212e-13      + chr20_43459420_43459770_+
 5: 20.172608     7 1.081055e-06 1.412099e-05      + chr20_43459420_43459770_+
 6:  8.475313     5 4.594393e-03 2.127349e-02      + chr20_43459420_43459770_+
 7: 26.926987     8 7.368617e-09 1.535400e-08      + chr20_43459420_43459770_+
 8: 16.263948     5 4.669469e-06 4.472329e-05      + chr20_43459420_43459770_+
 9: 65.386372     7 4.333776e-25 4.080603e-22      + chr20_43459420_43459770_+
10: 15.432546     7 6.583855e-05 1.207907e-03      + chr20_43459420_43459770_+
    score splice_site acceptors_skipped exons_skipped donors_skipped anchor
    &lt;int&gt;      &lt;char&gt;             &lt;int&gt;         &lt;int&gt;          &lt;int&gt; &lt;char&gt;
 1: 29930       GT-AG                 0             0              0      A
 2: 29930       GT-AG                 0             0              0      A
 3: 29930       GT-AG                 0             0              0      A
 4: 29930       GT-AG                 0             0              0      A
 5: 29930       GT-AG                 0             0              0      A
 6: 29930       GT-AG                 0             0              0      A
 7: 29930       GT-AG                 0             0              0      A
 8: 29930       GT-AG                 0             0              0      A
 9: 29930       GT-AG                 0             0              0      A
10: 29930       GT-AG                 0             0              0      A
    known_donor known_acceptor known_junction            gene_names
          &lt;int&gt;          &lt;int&gt;          &lt;int&gt;                &lt;char&gt;
 1:           0              1              0 ENSG00000288000,SRSF6
 2:           0              1              0 ENSG00000288000,SRSF6
 3:           0              1              0 ENSG00000288000,SRSF6
 4:           0              1              0 ENSG00000288000,SRSF6
 5:           0              1              0 ENSG00000288000,SRSF6
 6:           0              1              0 ENSG00000288000,SRSF6
 7:           0              1              0 ENSG00000288000,SRSF6
 8:           0              1              0 ENSG00000288000,SRSF6
 9:           0              1              0 ENSG00000288000,SRSF6
10:           0              1              0 ENSG00000288000,SRSF6
                                gene_ids
                                  &lt;char&gt;
 1: ENSG00000288000.1,ENSG00000124193.16
 2: ENSG00000288000.1,ENSG00000124193.16
 3: ENSG00000288000.1,ENSG00000124193.16
 4: ENSG00000288000.1,ENSG00000124193.16
 5: ENSG00000288000.1,ENSG00000124193.16
 6: ENSG00000288000.1,ENSG00000124193.16
 7: ENSG00000288000.1,ENSG00000124193.16
 8: ENSG00000288000.1,ENSG00000124193.16
 9: ENSG00000288000.1,ENSG00000124193.16
10: ENSG00000288000.1,ENSG00000124193.16
                                                                                  transcripts
                                                                                       &lt;char&gt;
 1: ENST00000244020.5,ENST00000657241.1,ENST00000662078.1,ENST00000668808.1,ENST00000670741.1
 2: ENST00000244020.5,ENST00000657241.1,ENST00000662078.1,ENST00000668808.1,ENST00000670741.1
 3: ENST00000244020.5,ENST00000657241.1,ENST00000662078.1,ENST00000668808.1,ENST00000670741.1
 4: ENST00000244020.5,ENST00000657241.1,ENST00000662078.1,ENST00000668808.1,ENST00000670741.1
 5: ENST00000244020.5,ENST00000657241.1,ENST00000662078.1,ENST00000668808.1,ENST00000670741.1
 6: ENST00000244020.5,ENST00000657241.1,ENST00000662078.1,ENST00000668808.1,ENST00000670741.1
 7: ENST00000244020.5,ENST00000657241.1,ENST00000662078.1,ENST00000668808.1,ENST00000670741.1
 8: ENST00000244020.5,ENST00000657241.1,ENST00000662078.1,ENST00000668808.1,ENST00000670741.1
 9: ENST00000244020.5,ENST00000657241.1,ENST00000662078.1,ENST00000668808.1,ENST00000670741.1
10: ENST00000244020.5,ENST00000657241.1,ENST00000662078.1,ENST00000668808.1,ENST00000670741.1
        5ss_seq    3ss_seq 5ss_score 3ss_score            Intron_coord
         &lt;char&gt;     &lt;char&gt;     &lt;num&gt;     &lt;num&gt;                  &lt;char&gt;
 1: TGATGTAAGTA TGTGATAAAG  8.411946  1.462716 chr20:43459420-43459771
 2: TGATGTAAGTA TGTGATAAAG  8.411946  1.462716 chr20:43459420-43459771
 3: TGATGTAAGTA TGTGATAAAG  8.411946  1.462716 chr20:43459420-43459771
 4: TGATGTAAGTA TGTGATAAAG  8.411946  1.462716 chr20:43459420-43459771
 5: TGATGTAAGTA TGTGATAAAG  8.411946  1.462716 chr20:43459420-43459771
 6: TGATGTAAGTA TGTGATAAAG  8.411946  1.462716 chr20:43459420-43459771
 7: TGATGTAAGTA TGTGATAAAG  8.411946  1.462716 chr20:43459420-43459771
 8: TGATGTAAGTA TGTGATAAAG  8.411946  1.462716 chr20:43459420-43459771
 9: TGATGTAAGTA TGTGATAAAG  8.411946  1.462716 chr20:43459420-43459771
10: TGATGTAAGTA TGTGATAAAG  8.411946  1.462716 chr20:43459420-43459771
    Gene_name  Annot Coding    UTR  IsSig
       &lt;char&gt; &lt;lgcl&gt; &lt;lgcl&gt; &lt;lgcl&gt; &lt;lgcl&gt;
 1:     SRSF6   TRUE  FALSE  FALSE   TRUE
 2:     SRSF6   TRUE  FALSE  FALSE   TRUE
 3:     SRSF6   TRUE  FALSE  FALSE   TRUE
 4:     SRSF6   TRUE  FALSE  FALSE   TRUE
 5:     SRSF6   TRUE  FALSE  FALSE   TRUE
 6:     SRSF6   TRUE  FALSE  FALSE  FALSE
 7:     SRSF6   TRUE  FALSE  FALSE   TRUE
 8:     SRSF6   TRUE  FALSE  FALSE   TRUE
 9:     SRSF6   TRUE  FALSE  FALSE   TRUE
10:     SRSF6   TRUE  FALSE  FALSE   TRUE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>DE.results <span class="sc">%&gt;%</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Symbol <span class="sc">==</span> <span class="st">"SRSF6"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">color=</span>condition, <span class="at">x=</span>logFC, <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(FDR))) <span class="sc">+</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.01</span>), <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="20250716_Explore_DifferentialExpressionSplicingResults_files/figure-html/unnamed-chunk-6-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Ok, that checks out… the chr20:43459420:43459771:clu_53323_+ is significantly upregulated in most all contrasts, except weirdly not UPF1</p>
<p>Let’s try some of the other SR proteins and splicing genes that are known conserved NMD targets…</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>DE.results <span class="sc">%&gt;%</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Symbol <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"U2AF2"</span>, <span class="st">"RBFOX2"</span>, <span class="st">"SRSF4"</span>, <span class="st">"SRSF3"</span>, <span class="st">"SRSF6"</span>, <span class="st">"HNRNPL"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">color=</span>condition, <span class="at">x=</span>logFC, <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(FDR))) <span class="sc">+</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.01</span>), <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>Symbol)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="20250716_Explore_DifferentialExpressionSplicingResults_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Ok not so sure what to make of those DE results… Will check splicing results for those genes…</p>
<p>And some other conserved ones mentioned in the literature… like PSD95 (DLG4, ENSG00000132535)</p>
<p>…</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>DE.results <span class="sc">%&gt;%</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="fu">str_detect</span>(GeneID, <span class="st">"ENSG00000132535"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">color=</span>condition, <span class="at">x=</span>logFC, <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(FDR))) <span class="sc">+</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.01</span>), <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="20250716_Explore_DifferentialExpressionSplicingResults_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>and bak1, though i’m not sure if the NMD event studied by sika zheng lab is conserved in humans</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>DE.results <span class="sc">%&gt;%</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Symbol <span class="sc">==</span> <span class="st">"BAK1"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">color=</span>condition, <span class="at">x=</span>logFC, <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(FDR))) <span class="sc">+</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.01</span>), <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="20250716_Explore_DifferentialExpressionSplicingResults_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Now let’s check ARHGAP17, an exon-skipping NMD event I highlighted in the leafcutter2 manuscript.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>diff.splicing.results <span class="sc">%&gt;%</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Gene_name<span class="sc">==</span><span class="st">"ARHGAP17"</span> <span class="sc">&amp;</span> <span class="sc">!</span>Coding) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                                      Contrast
                                                        &lt;char&gt;
1:                           Colombo.SM67_and_SMG7_vs_Scramble
2:                               Darman_Cycloxhexamide_vs_DMSO
3:     DylanUnpublishedFractionation.LCLNuclear_vs_Cytoplasmic
4: DylanUnpublishedFractionation.SH-SY5YNuclear_vs_Cytoplasmic
5:                        FairAndGEUVADIS_naRNA_vs_SteadyState
6:                        FairAndGEUVADIS_naRNA_vs_SteadyState
7:                        FairAndGEUVADIS_naRNA_vs_SteadyState
8:                    Ritter.ESCLightPolysome_vs_HeavyPolysome
9:                    Ritter.NPCLightPolysome_vs_HeavyPolysome
                                intron  chrom    start      end
                                &lt;char&gt; &lt;char&gt;    &lt;int&gt;    &lt;int&gt;
1: chr16:24970580:24972596:clu_29745_-  chr16 24970580 24972596
2: chr16:24954730:24959911:clu_29742_-  chr16 24954730 24959911
3: chr16:24954730:24959911:clu_29742_-  chr16 24954730 24959911
4: chr16:24954730:24959911:clu_29742_-  chr16 24954730 24959911
5: chr16:24954730:24959911:clu_29742_-  chr16 24954730 24959911
6: chr16:24970580:24972596:clu_29745_-  chr16 24970580 24972596
7: chr16:24972633:24977215:clu_29745_-  chr16 24972633 24977215
8: chr16:24954730:24959911:clu_29742_-  chr16 24954730 24959911
9: chr16:24954730:24959911:clu_29742_-  chr16 24954730 24959911
             cluster      logef      Control NMD_enriched    deltapsi  status
              &lt;char&gt;      &lt;num&gt;        &lt;num&gt;        &lt;num&gt;       &lt;num&gt;  &lt;char&gt;
1: chr16:clu_29745_-  0.3890447 3.491028e-03  0.007569930  0.40789025 Success
2: chr16:clu_29742_- -0.2263197 3.221865e-03  0.001649848 -0.15720170 Success
3: chr16:clu_29742_- -0.3556787 1.034933e-02  0.010134086 -0.02152450 Success
4: chr16:clu_29742_- -0.3565413 7.697497e-03  0.009329653  0.16321564 Success
5: chr16:clu_29742_- -0.6572197 3.375194e-03  0.002789473 -0.05857213 Success
6: chr16:clu_29745_-  0.4382338 4.213245e-04  0.021174854  2.07535296 Success
7: chr16:clu_29745_-  5.1917601 2.710970e-06  0.015803695  1.58009840 Success
8: chr16:clu_29742_-  1.1779676 5.410032e-04  0.003158253  0.26172493 Success
9: chr16:clu_29742_-  0.6285185 5.489159e-03  0.018104784  1.26156249 Success
        loglr    df            p     p.adjust strand                      name
        &lt;num&gt; &lt;int&gt;        &lt;num&gt;        &lt;num&gt; &lt;char&gt;                    &lt;char&gt;
1:  0.1941978     1 5.331439e-01 6.203317e-01      - chr16_24970580_24972595_-
2: 11.0422506     4 1.928054e-04 8.268606e-04      - chr16_24954730_24959910_-
3:  3.8453415     6 2.616517e-01 3.555646e-01      - chr16_24954730_24959910_-
4:  3.3291597     5 2.473166e-01 3.782919e-01      - chr16_24954730_24959910_-
5: 23.3859874     6 2.077663e-08 4.224925e-08      - chr16_24954730_24959910_-
6: 27.1918345     3 9.293664e-12 2.247543e-11      - chr16_24970580_24972595_-
7: 27.1918345     3 9.293664e-12 2.247543e-11      - chr16_24972633_24977214_-
8:  0.7080483     2 4.926047e-01 6.415110e-01      - chr16_24954730_24959910_-
9:  3.3533820     5 2.433779e-01 4.505200e-01      - chr16_24954730_24959910_-
   score splice_site acceptors_skipped exons_skipped donors_skipped anchor
   &lt;int&gt;      &lt;char&gt;             &lt;int&gt;         &lt;int&gt;          &lt;int&gt; &lt;char&gt;
1:   199       GT-AG                 0             0              0      A
2:   363       GT-AG                 1             1              1    NDA
3:   363       GT-AG                 1             1              1    NDA
4:   363       GT-AG                 1             1              1    NDA
5:   363       GT-AG                 1             1              1    NDA
6:   199       GT-AG                 0             0              0      A
7:   106       GT-AG                 0             0              0      D
8:   363       GT-AG                 1             1              1    NDA
9:   363       GT-AG                 1             1              1    NDA
   known_donor known_acceptor known_junction gene_names           gene_ids
         &lt;int&gt;          &lt;int&gt;          &lt;int&gt;     &lt;char&gt;             &lt;char&gt;
1:           0              1              0   ARHGAP17 ENSG00000140750.17
2:           1              1              0   ARHGAP17 ENSG00000140750.17
3:           1              1              0   ARHGAP17 ENSG00000140750.17
4:           1              1              0   ARHGAP17 ENSG00000140750.17
5:           1              1              0   ARHGAP17 ENSG00000140750.17
6:           0              1              0   ARHGAP17 ENSG00000140750.17
7:           1              0              0   ARHGAP17 ENSG00000140750.17
8:           1              1              0   ARHGAP17 ENSG00000140750.17
9:           1              1              0   ARHGAP17 ENSG00000140750.17
                            transcripts     5ss_seq    3ss_seq 5ss_score
                                 &lt;char&gt;      &lt;char&gt;     &lt;char&gt;     &lt;num&gt;
1: ENST00000289968.11,ENST00000303665.9 AAAGGTAGGAA TATCTTCCAG  9.315023
2: ENST00000289968.11,ENST00000303665.9 TACGGTAAGCA GCTGCTCTAG  8.374299
3: ENST00000289968.11,ENST00000303665.9 TACGGTAAGCA GCTGCTCTAG  8.374299
4: ENST00000289968.11,ENST00000303665.9 TACGGTAAGCA GCTGCTCTAG  8.374299
5: ENST00000289968.11,ENST00000303665.9 TACGGTAAGCA GCTGCTCTAG  8.374299
6: ENST00000289968.11,ENST00000303665.9 AAAGGTAGGAA TATCTTCCAG  9.315023
7: ENST00000289968.11,ENST00000303665.9 ACACGTGAGTA TCTTTAATAG  7.507509
8: ENST00000289968.11,ENST00000303665.9 TACGGTAAGCA GCTGCTCTAG  8.374299
9: ENST00000289968.11,ENST00000303665.9 TACGGTAAGCA GCTGCTCTAG  8.374299
   3ss_score            Intron_coord Gene_name  Annot Coding    UTR  IsSig
       &lt;num&gt;                  &lt;char&gt;    &lt;char&gt; &lt;lgcl&gt; &lt;lgcl&gt; &lt;lgcl&gt; &lt;lgcl&gt;
1:  8.726866 chr16:24970580-24972596  ARHGAP17   TRUE  FALSE   TRUE  FALSE
2:  4.224128 chr16:24954730-24959911  ARHGAP17   TRUE  FALSE  FALSE  FALSE
3:  4.224128 chr16:24954730-24959911  ARHGAP17   TRUE  FALSE  FALSE  FALSE
4:  4.224128 chr16:24954730-24959911  ARHGAP17   TRUE  FALSE  FALSE  FALSE
5:  4.224128 chr16:24954730-24959911  ARHGAP17   TRUE  FALSE  FALSE  FALSE
6:  8.726866 chr16:24970580-24972596  ARHGAP17   TRUE  FALSE   TRUE   TRUE
7:  6.968430 chr16:24972633-24977215  ARHGAP17   TRUE  FALSE  FALSE   TRUE
8:  4.224128 chr16:24954730-24959911  ARHGAP17   TRUE  FALSE  FALSE  FALSE
9:  4.224128 chr16:24954730-24959911  ARHGAP17   TRUE  FALSE  FALSE  FALSE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>diff.splicing.results <span class="sc">%&gt;%</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Gene_name<span class="sc">==</span><span class="st">"ARHGAP17"</span> <span class="sc">&amp;</span> <span class="sc">!</span>Coding) <span class="sc">%&gt;%</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">color=</span>Contrast, <span class="at">x=</span>deltapsi, <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(p.adjust))) <span class="sc">+</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.01</span>), <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>intron) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="20250716_Explore_DifferentialExpressionSplicingResults_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>diff.splicing.results <span class="sc">%&gt;%</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Gene_name<span class="sc">==</span><span class="st">"ARHGAP17"</span> <span class="sc">&amp;</span> <span class="sc">!</span>Coding) <span class="sc">%&gt;%</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(intron <span class="sc">==</span> <span class="st">"chr16:24954730:24959911:clu_29742_-"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                                      Contrast
                                                        &lt;char&gt;
1:                               Darman_Cycloxhexamide_vs_DMSO
2:     DylanUnpublishedFractionation.LCLNuclear_vs_Cytoplasmic
3: DylanUnpublishedFractionation.SH-SY5YNuclear_vs_Cytoplasmic
4:                        FairAndGEUVADIS_naRNA_vs_SteadyState
5:                    Ritter.ESCLightPolysome_vs_HeavyPolysome
6:                    Ritter.NPCLightPolysome_vs_HeavyPolysome
                                intron  chrom    start      end
                                &lt;char&gt; &lt;char&gt;    &lt;int&gt;    &lt;int&gt;
1: chr16:24954730:24959911:clu_29742_-  chr16 24954730 24959911
2: chr16:24954730:24959911:clu_29742_-  chr16 24954730 24959911
3: chr16:24954730:24959911:clu_29742_-  chr16 24954730 24959911
4: chr16:24954730:24959911:clu_29742_-  chr16 24954730 24959911
5: chr16:24954730:24959911:clu_29742_-  chr16 24954730 24959911
6: chr16:24954730:24959911:clu_29742_-  chr16 24954730 24959911
             cluster      logef      Control NMD_enriched    deltapsi  status
              &lt;char&gt;      &lt;num&gt;        &lt;num&gt;        &lt;num&gt;       &lt;num&gt;  &lt;char&gt;
1: chr16:clu_29742_- -0.2263197 0.0032218653  0.001649848 -0.15720170 Success
2: chr16:clu_29742_- -0.3556787 0.0103493312  0.010134086 -0.02152450 Success
3: chr16:clu_29742_- -0.3565413 0.0076974966  0.009329653  0.16321564 Success
4: chr16:clu_29742_- -0.6572197 0.0033751942  0.002789473 -0.05857213 Success
5: chr16:clu_29742_-  1.1779676 0.0005410032  0.003158253  0.26172493 Success
6: chr16:clu_29742_-  0.6285185 0.0054891587  0.018104784  1.26156249 Success
        loglr    df            p     p.adjust strand                      name
        &lt;num&gt; &lt;int&gt;        &lt;num&gt;        &lt;num&gt; &lt;char&gt;                    &lt;char&gt;
1: 11.0422506     4 1.928054e-04 8.268606e-04      - chr16_24954730_24959910_-
2:  3.8453415     6 2.616517e-01 3.555646e-01      - chr16_24954730_24959910_-
3:  3.3291597     5 2.473166e-01 3.782919e-01      - chr16_24954730_24959910_-
4: 23.3859874     6 2.077663e-08 4.224925e-08      - chr16_24954730_24959910_-
5:  0.7080483     2 4.926047e-01 6.415110e-01      - chr16_24954730_24959910_-
6:  3.3533820     5 2.433779e-01 4.505200e-01      - chr16_24954730_24959910_-
   score splice_site acceptors_skipped exons_skipped donors_skipped anchor
   &lt;int&gt;      &lt;char&gt;             &lt;int&gt;         &lt;int&gt;          &lt;int&gt; &lt;char&gt;
1:   363       GT-AG                 1             1              1    NDA
2:   363       GT-AG                 1             1              1    NDA
3:   363       GT-AG                 1             1              1    NDA
4:   363       GT-AG                 1             1              1    NDA
5:   363       GT-AG                 1             1              1    NDA
6:   363       GT-AG                 1             1              1    NDA
   known_donor known_acceptor known_junction gene_names           gene_ids
         &lt;int&gt;          &lt;int&gt;          &lt;int&gt;     &lt;char&gt;             &lt;char&gt;
1:           1              1              0   ARHGAP17 ENSG00000140750.17
2:           1              1              0   ARHGAP17 ENSG00000140750.17
3:           1              1              0   ARHGAP17 ENSG00000140750.17
4:           1              1              0   ARHGAP17 ENSG00000140750.17
5:           1              1              0   ARHGAP17 ENSG00000140750.17
6:           1              1              0   ARHGAP17 ENSG00000140750.17
                            transcripts     5ss_seq    3ss_seq 5ss_score
                                 &lt;char&gt;      &lt;char&gt;     &lt;char&gt;     &lt;num&gt;
1: ENST00000289968.11,ENST00000303665.9 TACGGTAAGCA GCTGCTCTAG  8.374299
2: ENST00000289968.11,ENST00000303665.9 TACGGTAAGCA GCTGCTCTAG  8.374299
3: ENST00000289968.11,ENST00000303665.9 TACGGTAAGCA GCTGCTCTAG  8.374299
4: ENST00000289968.11,ENST00000303665.9 TACGGTAAGCA GCTGCTCTAG  8.374299
5: ENST00000289968.11,ENST00000303665.9 TACGGTAAGCA GCTGCTCTAG  8.374299
6: ENST00000289968.11,ENST00000303665.9 TACGGTAAGCA GCTGCTCTAG  8.374299
   3ss_score            Intron_coord Gene_name  Annot Coding    UTR  IsSig
       &lt;num&gt;                  &lt;char&gt;    &lt;char&gt; &lt;lgcl&gt; &lt;lgcl&gt; &lt;lgcl&gt; &lt;lgcl&gt;
1:  4.224128 chr16:24954730-24959911  ARHGAP17   TRUE  FALSE  FALSE  FALSE
2:  4.224128 chr16:24954730-24959911  ARHGAP17   TRUE  FALSE  FALSE  FALSE
3:  4.224128 chr16:24954730-24959911  ARHGAP17   TRUE  FALSE  FALSE  FALSE
4:  4.224128 chr16:24954730-24959911  ARHGAP17   TRUE  FALSE  FALSE  FALSE
5:  4.224128 chr16:24954730-24959911  ARHGAP17   TRUE  FALSE  FALSE  FALSE
6:  4.224128 chr16:24954730-24959911  ARHGAP17   TRUE  FALSE  FALSE  FALSE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>DE.results <span class="sc">%&gt;%</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Symbol <span class="sc">==</span> <span class="st">"ARHGAP17"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">color=</span>condition, <span class="at">x=</span>logFC, <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(FDR))) <span class="sc">+</span></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.01</span>), <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="20250716_Explore_DifferentialExpressionSplicingResults_files/figure-html/unnamed-chunk-10-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Ok this one is interesting… I had to manually look up the coordinates of the exon skipping junction I mentioned in leafcutter2 manuscript, as chr16:24954730:24959911:clu_29742_-, but not sensible effect size direction in the cyclohexamide vs dmso contrast or naRNA vs steady state… Not sure what to make of this one… And the effect size of the host gene expression is not significant or negative (which could still be consistent with NMD junctions, if they are very rarely used).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>diff.splicing.results <span class="sc">%&gt;%</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Gene_name<span class="sc">==</span><span class="st">"SCN8A"</span> <span class="sc">&amp;</span> <span class="sc">!</span>Coding)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                   Contrast                              intron
                                     &lt;char&gt;                              &lt;char&gt;
1: Ritter.NPCLightPolysome_vs_HeavyPolysome chr12:51766027:51768113:clu_17447_+
2: Ritter.NPCLightPolysome_vs_HeavyPolysome chr12:51768172:51768865:clu_17447_+
3: Ritter.NPCLightPolysome_vs_HeavyPolysome chr12:51774362:51780202:clu_17449_+
4: Ritter.NPCLightPolysome_vs_HeavyPolysome chr12:51780271:51786542:clu_17449_+
    chrom    start      end           cluster       logef      Control
   &lt;char&gt;    &lt;int&gt;    &lt;int&gt;            &lt;char&gt;       &lt;num&gt;        &lt;num&gt;
1:  chr12 51766027 51768113 chr12:clu_17447_+  7.09371777 3.112383e-07
2:  chr12 51768172 51768865 chr12:clu_17447_+ -3.54685889 1.312997e-02
3:  chr12 51774362 51780202 chr12:clu_17449_+  0.09507781 3.433445e-01
4:  chr12 51780271 51786542 chr12:clu_17449_+  0.08507083 3.259497e-01
   NMD_enriched    deltapsi  status     loglr    df         p  p.adjust strand
          &lt;num&gt;       &lt;num&gt;  &lt;char&gt;     &lt;num&gt; &lt;int&gt;     &lt;num&gt;     &lt;num&gt; &lt;char&gt;
1:   0.01284174  1.28414245 Success 1.2942240     2 0.2741105 0.4853404      +
2:   0.01296136 -0.01686075 Success 1.2942240     2 0.2741105 0.4853404      +
3:   0.37434529  3.10007600 Success 0.2670796     2 0.7656122 0.8846779      +
4:   0.35184130  2.58916237 Success 0.2670796     2 0.7656122 0.8846779      +
                        name score splice_site acceptors_skipped exons_skipped
                      &lt;char&gt; &lt;int&gt;      &lt;char&gt;             &lt;int&gt;         &lt;int&gt;
1: chr12_51766027_51768112_+    35       GT-AG                 0             0
2: chr12_51768172_51768864_+    49       GT-AG                 0             0
3: chr12_51774362_51780201_+   661       GT-AG                 0             0
4: chr12_51780271_51786541_+   591       GT-AG                 1             1
   donors_skipped anchor known_donor known_acceptor known_junction gene_names
            &lt;int&gt; &lt;char&gt;       &lt;int&gt;          &lt;int&gt;          &lt;int&gt;     &lt;char&gt;
1:              0      D           1              0              0      SCN8A
2:              0      A           0              1              0      SCN8A
3:              0      D           1              0              0      SCN8A
4:              1      A           0              1              0      SCN8A
             gene_ids
               &lt;char&gt;
1: ENSG00000196876.19
2: ENSG00000196876.19
3: ENSG00000196876.19
4: ENSG00000196876.19
                                                                                                                      transcripts
                                                                                                                           &lt;char&gt;
1: ENST00000354534.11,ENST00000355133.7,ENST00000545061.5,ENST00000599343.5,ENST00000627620.5,ENST00000662684.1,ENST00000668547.1
2: ENST00000354534.11,ENST00000355133.7,ENST00000545061.5,ENST00000599343.5,ENST00000627620.5,ENST00000662684.1,ENST00000668547.1
3: ENST00000354534.11,ENST00000355133.7,ENST00000545061.5,ENST00000599343.5,ENST00000627620.5,ENST00000662684.1,ENST00000668547.1
4: ENST00000354534.11,ENST00000355133.7,ENST00000545061.5,ENST00000599343.5,ENST00000627620.5,ENST00000662684.1,ENST00000668547.1
       5ss_seq    3ss_seq 5ss_score 3ss_score            Intron_coord Gene_name
        &lt;char&gt;     &lt;char&gt;     &lt;num&gt;     &lt;num&gt;                  &lt;char&gt;    &lt;char&gt;
1: GGTGGTTAGTA TCTTCCCAAG  5.793933 5.8097940 chr12:51766027-51768113     SCN8A
2: AAAAGTAAGTC ACTGGCATAG  9.380309 0.8051234 chr12:51768172-51768865     SCN8A
3: GGCTGTAGGTG TTTTCCATAG  3.451919 8.5187230 chr12:51774362-51780202     SCN8A
4: GAGGGTAAGGC TCCAATGCAG  8.192756 5.3371230 chr12:51780271-51786542     SCN8A
    Annot Coding    UTR  IsSig
   &lt;lgcl&gt; &lt;lgcl&gt; &lt;lgcl&gt; &lt;lgcl&gt;
1:   TRUE  FALSE  FALSE  FALSE
2:   TRUE  FALSE  FALSE  FALSE
3:   TRUE  FALSE  FALSE  FALSE
4:   TRUE  FALSE  FALSE  FALSE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>diff.splicing.results <span class="sc">%&gt;%</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Gene_name<span class="sc">==</span><span class="st">"SCN8A"</span> <span class="sc">&amp;</span> <span class="sc">!</span>Coding) <span class="sc">%&gt;%</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">color=</span>Contrast, <span class="at">x=</span>deltapsi, <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(p.adjust))) <span class="sc">+</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.01</span>), <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>intron) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="20250716_Explore_DifferentialExpressionSplicingResults_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>DE.results <span class="sc">%&gt;%</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Symbol <span class="sc">==</span> <span class="st">"SCN8A"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">color=</span>condition, <span class="at">x=</span>logFC, <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(FDR))) <span class="sc">+</span></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.01</span>), <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="20250716_Explore_DifferentialExpressionSplicingResults_files/figure-html/unnamed-chunk-11-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>no splicing tests in SCN8A… I think it is just not expressed in any of these samples. But there are tests in differential expression that suggest up-regulation in nuclear RNA… I could recover that significant splice junction in my re-analysis of Mazin data in leafcutter2 manuscript. I could consider adding all that data to the analysis.</p>
<p>Let’s try SYNGAP1</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>diff.splicing.results <span class="sc">%&gt;%</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Gene_name<span class="sc">==</span><span class="st">"SYNGAP1"</span> <span class="sc">&amp;</span> <span class="sc">!</span>Coding) <span class="sc">%&gt;%</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">color=</span>Contrast, <span class="at">x=</span>deltapsi, <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(p.adjust))) <span class="sc">+</span></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.01</span>), <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>intron) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="20250716_Explore_DifferentialExpressionSplicingResults_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>diff.splicing.results <span class="sc">%&gt;%</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Gene_name<span class="sc">==</span><span class="st">"SYNGAP1"</span> <span class="sc">&amp;</span> <span class="sc">!</span>Coding) <span class="sc">%&gt;%</span></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(intron <span class="sc">==</span> <span class="st">"chr6:33438919:33440553:clu_72389_+"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                                      Contrast
                                                        &lt;char&gt;
1:                           Colombo.SM67_and_SMG7_vs_Scramble
2:                                    Colombo.UPF1_vs_Scramble
3:                               Darman_Cycloxhexamide_vs_DMSO
4:  DylanUnpublishedFractionation.JurkatNuclear_vs_Cytoplasmic
5: DylanUnpublishedFractionation.SH-SY5YNuclear_vs_Cytoplasmic
6:                        FairAndGEUVADIS_naRNA_vs_SteadyState
7:                    Ritter.ESCLightPolysome_vs_HeavyPolysome
8:                    Ritter.NPCLightPolysome_vs_HeavyPolysome
                               intron  chrom    start      end          cluster
                               &lt;char&gt; &lt;char&gt;    &lt;int&gt;    &lt;int&gt;           &lt;char&gt;
1: chr6:33438919:33440553:clu_72389_+   chr6 33438919 33440553 chr6:clu_72389_+
2: chr6:33438919:33440553:clu_72389_+   chr6 33438919 33440553 chr6:clu_72389_+
3: chr6:33438919:33440553:clu_72389_+   chr6 33438919 33440553 chr6:clu_72389_+
4: chr6:33438919:33440553:clu_72389_+   chr6 33438919 33440553 chr6:clu_72389_+
5: chr6:33438919:33440553:clu_72389_+   chr6 33438919 33440553 chr6:clu_72389_+
6: chr6:33438919:33440553:clu_72389_+   chr6 33438919 33440553 chr6:clu_72389_+
7: chr6:33438919:33440553:clu_72389_+   chr6 33438919 33440553 chr6:clu_72389_+
8: chr6:33438919:33440553:clu_72389_+   chr6 33438919 33440553 chr6:clu_72389_+
        logef   Control NMD_enriched   deltapsi  status     loglr    df
        &lt;num&gt;     &lt;num&gt;        &lt;num&gt;      &lt;num&gt;  &lt;char&gt;     &lt;num&gt; &lt;int&gt;
1:  2.0153414 0.4737337    0.5916376 11.7903890 Success  3.870732     5
2:  0.5196368 0.4735339    0.6528797 17.9345721 Success  3.494346     5
3:  0.4956944 0.3810956    0.6357550 25.4659365 Success 11.448234     3
4: -1.3885964 0.2750760    0.5352371 26.0161104 Success 10.749786     5
5:  0.5317421 0.2743841    0.3976508 12.3266707 Success  7.145558     5
6: -0.4181101 0.3594852    0.3520050 -0.7480132 Success 12.324268     6
7:  1.7773455 0.3329563    0.4483756 11.5419251 Success 13.287976     6
8:  0.2779642 0.3198095    0.3640423  4.4232717 Success 15.447554     6
              p     p.adjust strand                     name score splice_site
          &lt;num&gt;        &lt;num&gt; &lt;char&gt;                   &lt;char&gt; &lt;int&gt;      &lt;char&gt;
1: 1.710715e-01 0.2441195859      + chr6_33438919_33440552_+  7891       GT-AG
2: 2.214827e-01 0.3435305018      + chr6_33438919_33440552_+  7891       GT-AG
3: 4.244012e-05 0.0002271387      + chr6_33438919_33440552_+  7891       GT-AG
4: 6.516057e-04 0.0033553019      + chr6_33438919_33440552_+  7891       GT-AG
5: 1.386221e-02 0.0485223318      + chr6_33438919_33440552_+  7891       GT-AG
6: 3.965835e-04 0.0006061113      + chr6_33438919_33440552_+  7891       GT-AG
7: 1.738356e-04 0.0017165937      + chr6_33438919_33440552_+  7891       GT-AG
8: 2.654530e-05 0.0005755812      + chr6_33438919_33440552_+  7891       GT-AG
   acceptors_skipped exons_skipped donors_skipped anchor known_donor
               &lt;int&gt;         &lt;int&gt;          &lt;int&gt; &lt;char&gt;       &lt;int&gt;
1:                 0             0              0      D           1
2:                 0             0              0      D           1
3:                 0             0              0      D           1
4:                 0             0              0      D           1
5:                 0             0              0      D           1
6:                 0             0              0      D           1
7:                 0             0              0      D           1
8:                 0             0              0      D           1
   known_acceptor known_junction gene_names           gene_ids
            &lt;int&gt;          &lt;int&gt;     &lt;char&gt;             &lt;char&gt;
1:              0              0    SYNGAP1 ENSG00000197283.18
2:              0              0    SYNGAP1 ENSG00000197283.18
3:              0              0    SYNGAP1 ENSG00000197283.18
4:              0              0    SYNGAP1 ENSG00000197283.18
5:              0              0    SYNGAP1 ENSG00000197283.18
6:              0              0    SYNGAP1 ENSG00000197283.18
7:              0              0    SYNGAP1 ENSG00000197283.18
8:              0              0    SYNGAP1 ENSG00000197283.18
                                                                                                                                                         transcripts
                                                                                                                                                              &lt;char&gt;
1: ENST00000418600.7,ENST00000428982.4,ENST00000449372.7,ENST00000628646.2,ENST00000629380.3,ENST00000644458.1,ENST00000645250.1,ENST00000646630.1,ENST00000682587.1
2: ENST00000418600.7,ENST00000428982.4,ENST00000449372.7,ENST00000628646.2,ENST00000629380.3,ENST00000644458.1,ENST00000645250.1,ENST00000646630.1,ENST00000682587.1
3: ENST00000418600.7,ENST00000428982.4,ENST00000449372.7,ENST00000628646.2,ENST00000629380.3,ENST00000644458.1,ENST00000645250.1,ENST00000646630.1,ENST00000682587.1
4: ENST00000418600.7,ENST00000428982.4,ENST00000449372.7,ENST00000628646.2,ENST00000629380.3,ENST00000644458.1,ENST00000645250.1,ENST00000646630.1,ENST00000682587.1
5: ENST00000418600.7,ENST00000428982.4,ENST00000449372.7,ENST00000628646.2,ENST00000629380.3,ENST00000644458.1,ENST00000645250.1,ENST00000646630.1,ENST00000682587.1
6: ENST00000418600.7,ENST00000428982.4,ENST00000449372.7,ENST00000628646.2,ENST00000629380.3,ENST00000644458.1,ENST00000645250.1,ENST00000646630.1,ENST00000682587.1
7: ENST00000418600.7,ENST00000428982.4,ENST00000449372.7,ENST00000628646.2,ENST00000629380.3,ENST00000644458.1,ENST00000645250.1,ENST00000646630.1,ENST00000682587.1
8: ENST00000418600.7,ENST00000428982.4,ENST00000449372.7,ENST00000628646.2,ENST00000629380.3,ENST00000644458.1,ENST00000645250.1,ENST00000646630.1,ENST00000682587.1
       5ss_seq    3ss_seq 5ss_score 3ss_score           Intron_coord Gene_name
        &lt;char&gt;     &lt;char&gt;     &lt;num&gt;     &lt;num&gt;                 &lt;char&gt;    &lt;char&gt;
1: ACTGGTGAGAC CTTCAAGCAG  8.165933   4.80902 chr6:33438919-33440553   SYNGAP1
2: ACTGGTGAGAC CTTCAAGCAG  8.165933   4.80902 chr6:33438919-33440553   SYNGAP1
3: ACTGGTGAGAC CTTCAAGCAG  8.165933   4.80902 chr6:33438919-33440553   SYNGAP1
4: ACTGGTGAGAC CTTCAAGCAG  8.165933   4.80902 chr6:33438919-33440553   SYNGAP1
5: ACTGGTGAGAC CTTCAAGCAG  8.165933   4.80902 chr6:33438919-33440553   SYNGAP1
6: ACTGGTGAGAC CTTCAAGCAG  8.165933   4.80902 chr6:33438919-33440553   SYNGAP1
7: ACTGGTGAGAC CTTCAAGCAG  8.165933   4.80902 chr6:33438919-33440553   SYNGAP1
8: ACTGGTGAGAC CTTCAAGCAG  8.165933   4.80902 chr6:33438919-33440553   SYNGAP1
    Annot Coding    UTR  IsSig
   &lt;lgcl&gt; &lt;lgcl&gt; &lt;lgcl&gt; &lt;lgcl&gt;
1:   TRUE  FALSE  FALSE  FALSE
2:   TRUE  FALSE  FALSE  FALSE
3:   TRUE  FALSE  FALSE   TRUE
4:   TRUE  FALSE  FALSE   TRUE
5:   TRUE  FALSE  FALSE  FALSE
6:   TRUE  FALSE  FALSE  FALSE
7:   TRUE  FALSE  FALSE   TRUE
8:   TRUE  FALSE  FALSE   TRUE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>DE.results <span class="sc">%&gt;%</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Symbol <span class="sc">==</span> <span class="st">"SYNGAP1"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">color=</span>condition, <span class="at">x=</span>logFC, <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(FDR))) <span class="sc">+</span></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.01</span>), <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="20250716_Explore_DifferentialExpressionSplicingResults_files/figure-html/unnamed-chunk-12-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Ok, generally a positive deltaPSI… again some disagreement between naRNA vs steady state and the other contrasts… that’s not necessarily unsurprising considering how unspliced naRNA is.</p>
<p>Let’s check some others that the TANGO paper mentions… SCN1A?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>diff.splicing.results <span class="sc">%&gt;%</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Gene_name<span class="sc">==</span><span class="st">"SCN1A"</span> <span class="sc">&amp;</span> <span class="sc">!</span>Coding)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Empty data.table (0 rows and 39 cols): Contrast,intron,chrom,start,end,cluster...</code></pre>
</div>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>DE.results <span class="sc">%&gt;%</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Symbol <span class="sc">==</span> <span class="st">"SCN1A"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">color=</span>condition, <span class="at">x=</span>logFC, <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(FDR))) <span class="sc">+</span></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.01</span>), <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="20250716_Explore_DifferentialExpressionSplicingResults_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Ok no tests for SCN1A… it just isn’t tested/expressed in these analyses.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>diff.splicing.results <span class="sc">%&gt;%</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Gene_name<span class="sc">==</span><span class="st">"CD274"</span> <span class="sc">&amp;</span> <span class="sc">!</span>Coding)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                                  Contrast
                                                    &lt;char&gt;
1: DylanUnpublishedFractionation.LCLNuclear_vs_Cytoplasmic
2:                    FairAndGEUVADIS_naRNA_vs_SteadyState
                             intron  chrom   start     end          cluster
                             &lt;char&gt; &lt;char&gt;   &lt;int&gt;   &lt;int&gt;           &lt;char&gt;
1: chr9:5465539:5466770:clu_83002_+   chr9 5465539 5466770 chr9:clu_83002_+
2: chr9:5465539:5466770:clu_83002_+   chr9 5465539 5466770 chr9:clu_83002_+
        logef      Control NMD_enriched   deltapsi  status     loglr    df
        &lt;num&gt;        &lt;num&gt;        &lt;num&gt;      &lt;num&gt;  &lt;char&gt;     &lt;num&gt; &lt;int&gt;
1: 6.38117467 6.275008e-07  0.008464249  0.8463622 Success  1.170946     2
2: 0.09641209 1.564953e-02  0.011924056 -0.3725470 Success 28.362063     4
              p     p.adjust strand                   name score splice_site
          &lt;num&gt;        &lt;num&gt; &lt;char&gt;                 &lt;char&gt; &lt;int&gt;      &lt;char&gt;
1: 3.100734e-01 4.054199e-01      + chr9_5465539_5466769_+   220       GT-AG
2: 1.413511e-11 3.389371e-11      + chr9_5465539_5466769_+   220       GT-AG
   acceptors_skipped exons_skipped donors_skipped anchor known_donor
               &lt;int&gt;         &lt;int&gt;          &lt;int&gt; &lt;char&gt;       &lt;int&gt;
1:                 0             0              1      A           0
2:                 0             0              1      A           0
   known_acceptor known_junction gene_names           gene_ids
            &lt;int&gt;          &lt;int&gt;     &lt;char&gt;             &lt;char&gt;
1:              1              0      CD274 ENSG00000120217.14
2:              1              0      CD274 ENSG00000120217.14
                           transcripts     5ss_seq    3ss_seq 5ss_score
                                &lt;char&gt;      &lt;char&gt;     &lt;char&gt;     &lt;num&gt;
1: ENST00000381573.8,ENST00000381577.4 CTTGGTAATTC TTTTCTCAAG  5.560873
2: ENST00000381573.8,ENST00000381577.4 CTTGGTAATTC TTTTCTCAAG  5.560873
   3ss_score         Intron_coord Gene_name  Annot Coding    UTR  IsSig
       &lt;num&gt;               &lt;char&gt;    &lt;char&gt; &lt;lgcl&gt; &lt;lgcl&gt; &lt;lgcl&gt; &lt;lgcl&gt;
1:  7.583448 chr9:5465539-5466770     CD274   TRUE  FALSE  FALSE  FALSE
2:  7.583448 chr9:5465539-5466770     CD274   TRUE  FALSE  FALSE  FALSE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>diff.splicing.results <span class="sc">%&gt;%</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Gene_name<span class="sc">==</span><span class="st">"CD274"</span> <span class="sc">&amp;</span> <span class="sc">!</span>Coding) <span class="sc">%&gt;%</span></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">color=</span>Contrast, <span class="at">x=</span>deltapsi, <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(p.adjust))) <span class="sc">+</span></span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.01</span>), <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>intron) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="20250716_Explore_DifferentialExpressionSplicingResults_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>DE.results <span class="sc">%&gt;%</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Symbol <span class="sc">==</span> <span class="st">"CD274"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">color=</span>condition, <span class="at">x=</span>logFC, <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(FDR))) <span class="sc">+</span></span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.01</span>), <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="20250716_Explore_DifferentialExpressionSplicingResults_files/figure-html/unnamed-chunk-14-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>And PCCA…</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>diff.splicing.results <span class="sc">%&gt;%</span></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Gene_name<span class="sc">==</span><span class="st">"PCCA"</span> <span class="sc">&amp;</span> <span class="sc">!</span>Coding) <span class="sc">%&gt;%</span></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">color=</span>Contrast, <span class="at">x=</span>deltapsi<span class="sc">*</span><span class="dv">100</span>, <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(p.adjust))) <span class="sc">+</span></span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.01</span>), <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>intron, <span class="at">scales=</span><span class="st">"free"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="20250716_Explore_DifferentialExpressionSplicingResults_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>diff.splicing.results <span class="sc">%&gt;%</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Gene_name<span class="sc">==</span><span class="st">"PCCA"</span> <span class="sc">&amp;</span> <span class="sc">!</span>Coding) <span class="sc">%&gt;%</span></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>Contrast <span class="sc">==</span> <span class="st">"FairAndGEUVADIS_naRNA_vs_SteadyState"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">color=</span>Contrast, <span class="at">x=</span>deltapsi, <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(p.adjust))) <span class="sc">+</span></span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.01</span>), <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb37-9"><a href="#cb37-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>intron) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="20250716_Explore_DifferentialExpressionSplicingResults_files/figure-html/unnamed-chunk-15-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>diff.splicing.results <span class="sc">%&gt;%</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Gene_name<span class="sc">==</span><span class="st">"PCCA"</span> <span class="sc">&amp;</span> <span class="sc">!</span>Coding) <span class="sc">%&gt;%</span></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>Contrast <span class="sc">==</span> <span class="st">"FairAndGEUVADIS_naRNA_vs_SteadyState"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">color=</span>Contrast, <span class="at">x=</span>deltapsi, <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(p))) <span class="sc">+</span></span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.01</span>), <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>intron) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="20250716_Explore_DifferentialExpressionSplicingResults_files/figure-html/unnamed-chunk-15-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>DE.results <span class="sc">%&gt;%</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Symbol <span class="sc">==</span> <span class="st">"PCCA"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">color=</span>condition, <span class="at">x=</span>logFC, <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(FDR))) <span class="sc">+</span></span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.01</span>), <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="20250716_Explore_DifferentialExpressionSplicingResults_files/figure-html/unnamed-chunk-15-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Ok from all of these, I think a trend might be emerging that gene expression might be a decent way to look at this problem… Let’s check that gene of interest…</p>
</section>
<section id="mbd5" class="level3">
<h3 class="anchored" data-anchor-id="mbd5">MBD5</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>DE.results <span class="sc">%&gt;%</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Symbol <span class="sc">==</span> <span class="st">"MBD5"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">color=</span>condition, <span class="at">x=</span>logFC, <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(FDR))) <span class="sc">+</span></span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.01</span>), <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="20250716_Explore_DifferentialExpressionSplicingResults_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>diff.splicing.results <span class="sc">%&gt;%</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Gene_name<span class="sc">==</span><span class="st">"MBD5"</span> <span class="sc">&amp;</span> <span class="sc">!</span>Coding) <span class="sc">%&gt;%</span></span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span>Contrast <span class="sc">==</span> <span class="st">"FairAndGEUVADIS_naRNA_vs_SteadyState"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">color=</span>Contrast, <span class="at">x=</span>deltapsi, <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(p))) <span class="sc">+</span></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.01</span>), <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>intron) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="20250716_Explore_DifferentialExpressionSplicingResults_files/figure-html/unnamed-chunk-16-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>diff.splicing.results <span class="sc">%&gt;%</span></span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Gene_name<span class="sc">==</span><span class="st">"MBD5"</span> <span class="sc">&amp;</span> <span class="sc">!</span>Coding) <span class="sc">%&gt;%</span></span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(deltapsi <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> p <span class="sc">&lt;</span> <span class="fl">0.01</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                                      Contrast
                                                        &lt;char&gt;
 1:                              Darman_Cycloxhexamide_vs_DMSO
 2:                              Darman_Cycloxhexamide_vs_DMSO
 3:                              Darman_Cycloxhexamide_vs_DMSO
 4: DylanUnpublishedFractionation.JurkatNuclear_vs_Cytoplasmic
 5: DylanUnpublishedFractionation.JurkatNuclear_vs_Cytoplasmic
 6:                       FairAndGEUVADIS_naRNA_vs_SteadyState
 7:                       FairAndGEUVADIS_naRNA_vs_SteadyState
 8:                       FairAndGEUVADIS_naRNA_vs_SteadyState
 9:                       FairAndGEUVADIS_naRNA_vs_SteadyState
10:                   Ritter.ESCLightPolysome_vs_HeavyPolysome
                                  intron  chrom     start       end
                                  &lt;char&gt; &lt;char&gt;     &lt;int&gt;     &lt;int&gt;
 1: chr2:148021684:148053933:clu_50423_+   chr2 148021684 148053933
 2: chr2:148021684:148054391:clu_50423_+   chr2 148021684 148054391
 3: chr2:148054050:148054391:clu_50423_+   chr2 148054050 148054391
 4: chr2:148021684:148054391:clu_50423_+   chr2 148021684 148054391
 5: chr2:148054449:148070873:clu_50423_+   chr2 148054449 148070873
 6: chr2:148021684:148053933:clu_50423_+   chr2 148021684 148053933
 7: chr2:148021684:148054391:clu_50423_+   chr2 148021684 148054391
 8: chr2:148054050:148054391:clu_50423_+   chr2 148054050 148054391
 9: chr2:148502509:148502965:clu_50435_+   chr2 148502509 148502965
10: chr2:148021684:148024966:clu_50423_+   chr2 148021684 148024966
             cluster      logef      Control NMD_enriched    deltapsi  status
              &lt;char&gt;      &lt;num&gt;        &lt;num&gt;        &lt;num&gt;       &lt;num&gt;  &lt;char&gt;
 1: chr2:clu_50423_+  7.5888538 1.973041e-06  0.058050784  5.80488110 Success
 2: chr2:clu_50423_+ -1.6741507 2.623554e-02  0.073230181  4.69946437 Success
 3: chr2:clu_50423_+ -1.0989327 1.310412e-02  0.065016289  5.19121646 Success
 4: chr2:clu_50423_+  2.4003564 6.665125e-02  0.223640485 15.69892349 Success
 5: chr2:clu_50423_+  2.1809966 9.325724e-03  0.025127998  1.58022743 Success
 6: chr2:clu_50423_+ -0.3675929 6.807745e-02  0.095638672  2.75612220 Success
 7: chr2:clu_50423_+ -0.2044755 2.390188e-01  0.395278516 15.62597631 Success
 8: chr2:clu_50423_+ -0.2567600 8.041408e-02  0.126210781  4.57967025 Success
 9: chr2:clu_50435_+ -0.1065373 3.565674e-03  0.004077064  0.05113897 Success
10: chr2:clu_50423_+  1.0939205 4.045588e-01  0.708226005 30.36671739 Success
         loglr    df            p     p.adjust strand
         &lt;num&gt; &lt;int&gt;        &lt;num&gt;        &lt;num&gt; &lt;char&gt;
 1:   7.271054     4 5.751515e-03 1.470129e-02      +
 2:   7.271054     4 5.751515e-03 1.470129e-02      +
 3:   7.271054     4 5.751515e-03 1.470129e-02      +
 4:  14.154932    11 2.901552e-03 1.115752e-02      +
 5:  14.154932    11 2.901552e-03 1.115752e-02      +
 6: 151.121836    20 2.813703e-52 2.721958e-51      +
 7: 151.121836    20 2.813703e-52 2.721958e-51      +
 8: 151.121836    20 2.813703e-52 2.721958e-51      +
 9:  32.540022     5 1.078699e-12 2.722122e-12      +
10:  13.725970     8 5.905039e-04 4.686068e-03      +
                          name score splice_site acceptors_skipped
                        &lt;char&gt; &lt;int&gt;      &lt;char&gt;             &lt;int&gt;
 1: chr2_148021684_148053932_+  1099       GT-AG                 0
 2: chr2_148021684_148054390_+  4197       GT-AG                 0
 3: chr2_148054050_148054390_+  1433       GT-AG                 0
 4: chr2_148021684_148054390_+  4197       GT-AG                 0
 5: chr2_148054449_148070872_+   254       GT-AG                 0
 6: chr2_148021684_148053932_+  1099       GT-AG                 0
 7: chr2_148021684_148054390_+  4197       GT-AG                 0
 8: chr2_148054050_148054390_+  1433       GT-AG                 0
 9: chr2_148502509_148502964_+    38       GT-AG                 0
10: chr2_148021684_148024965_+  5080       GT-AG                 0
    exons_skipped donors_skipped anchor known_donor known_acceptor
            &lt;int&gt;          &lt;int&gt; &lt;char&gt;       &lt;int&gt;          &lt;int&gt;
 1:             0              0      D           1              0
 2:             0              0      D           1              0
 3:             0              0      N           0              0
 4:             0              0      D           1              0
 5:             0              0      N           0              0
 6:             0              0      D           1              0
 7:             0              0      D           1              0
 8:             0              0      N           0              0
 9:             0              0      D           1              0
10:             0              0      D           1              0
    known_junction gene_names           gene_ids
             &lt;int&gt;     &lt;char&gt;             &lt;char&gt;
 1:              0       MBD5 ENSG00000204406.14
 2:              0       MBD5 ENSG00000204406.14
 3:              0                              
 4:              0       MBD5 ENSG00000204406.14
 5:              0                              
 6:              0       MBD5 ENSG00000204406.14
 7:              0       MBD5 ENSG00000204406.14
 8:              0                              
 9:              0       MBD5 ENSG00000204406.14
10:              0       MBD5 ENSG00000204406.14
                                              transcripts     5ss_seq
                                                   &lt;char&gt;      &lt;char&gt;
 1: ENST00000407073.5,ENST00000638043.2,ENST00000642680.2 AAAGGTAAGTG
 2: ENST00000407073.5,ENST00000638043.2,ENST00000642680.2 AAAGGTAAGTG
 3:                                                       ACAGGTGGGCG
 4: ENST00000407073.5,ENST00000638043.2,ENST00000642680.2 AAAGGTAAGTG
 5:                                                       CACAGTAAGTT
 6: ENST00000407073.5,ENST00000638043.2,ENST00000642680.2 AAAGGTAAGTG
 7: ENST00000407073.5,ENST00000638043.2,ENST00000642680.2 AAAGGTAAGTG
 8:                                                       ACAGGTGGGCG
 9: ENST00000407073.5,ENST00000638043.2,ENST00000642680.2 GAAAGTAAGCA
10: ENST00000407073.5,ENST00000638043.2,ENST00000642680.2 AAAGGTAAGTG
       3ss_seq 5ss_score 3ss_score             Intron_coord Gene_name  Annot
        &lt;char&gt;     &lt;num&gt;     &lt;num&gt;                   &lt;char&gt;    &lt;char&gt; &lt;lgcl&gt;
 1: TCTATGACAG 13.040886  5.722426 chr2:148021684-148053933      MBD5   TRUE
 2: TGCTTAACAG 13.040886  6.208464 chr2:148021684-148054391      MBD5   TRUE
 3: TGCTTAACAG  8.322879  6.208464 chr2:148054050-148054391      MBD5   TRUE
 4: TGCTTAACAG 13.040886  6.208464 chr2:148021684-148054391      MBD5   TRUE
 5: TTTTTTTTAG  6.850367 10.348467 chr2:148054449-148070873      MBD5   TRUE
 6: TCTATGACAG 13.040886  5.722426 chr2:148021684-148053933      MBD5   TRUE
 7: TGCTTAACAG 13.040886  6.208464 chr2:148021684-148054391      MBD5   TRUE
 8: TGCTTAACAG  8.322879  6.208464 chr2:148054050-148054391      MBD5   TRUE
 9: GTTGCATCAG  7.955293  3.899733 chr2:148502509-148502965      MBD5   TRUE
10: CCTTACCCAG 13.040886  6.395020 chr2:148021684-148024966      MBD5   TRUE
    Coding    UTR  IsSig
    &lt;lgcl&gt; &lt;lgcl&gt; &lt;lgcl&gt;
 1:  FALSE   TRUE  FALSE
 2:  FALSE   TRUE  FALSE
 3:  FALSE  FALSE  FALSE
 4:  FALSE   TRUE  FALSE
 5:  FALSE  FALSE  FALSE
 6:  FALSE   TRUE   TRUE
 7:  FALSE   TRUE   TRUE
 8:  FALSE  FALSE   TRUE
 9:  FALSE  FALSE  FALSE
10:  FALSE   TRUE   TRUE</code></pre>
</div>
</div>
<p>This intron: chr2:148021684:148054391:clu_50423_+, seems most promising… but it is in the 5’ UTR region which I think is not good candidate for the specific case.</p>
<p>Also, note that later I think some of these polysome profiling could be useful to others in Hastings lab interested in splicing in 5UTR that may alter translation output.</p>
</section>
<section id="utr-splicing" class="level3">
<h3 class="anchored" data-anchor-id="utr-splicing">5’ UTR splicing</h3>
<p>Let’s filiter differential splicing results for introns that are in 5’UTR regions.</p>
<p>First get positions of 5’UTR for mane isoforms.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O-</span> <span class="st">"https://ftp.ncbi.nlm.nih.gov/refseq/MANE/MANE_human/current/MANE.GRCh38.v1.4.ensembl_genomic.gtf.gz"</span> <span class="kw">|</span> <span class="fu">zcat</span> <span class="kw">|</span> <span class="ex">bedparse</span> gtf2bed <span class="at">--extraFields</span> gene_id,gene_name /dev/stdin <span class="kw">|</span> <span class="fu">awk</span> <span class="at">-v</span> OFS=<span class="st">'\t'</span> <span class="st">'{$4=$4"_"$13"_"$14; print $1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12}'</span> <span class="kw">|</span> <span class="ex">bedparse</span> 5pUTR <span class="kw">|</span> <span class="fu">gzip</span> <span class="at">-</span> <span class="op">&gt;</span> ../data/5pUTR.bed.gz</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now read in the 5’UTR positions</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>fiveUTR <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"../data/5pUTR.bed.gz"</span>, <span class="at">col_names =</span> <span class="fu">c</span>(<span class="st">"chrom"</span>, <span class="st">"start"</span>, <span class="st">"end"</span>, <span class="st">"name"</span>, <span class="st">"score"</span>, <span class="st">"strand"</span>, <span class="st">"thickStart"</span>, <span class="st">"thickEnd"</span>, <span class="st">"color"</span>, <span class="st">"nBlocks"</span>, <span class="st">"blockSizes"</span>, <span class="st">"blockStarts"</span>))</span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a>fiveUTR <span class="sc">%&gt;%</span> <span class="fu">head</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 12
  chrom    start      end name    score strand thickStart thickEnd color nBlocks
  &lt;chr&gt;    &lt;dbl&gt;    &lt;dbl&gt; &lt;chr&gt;   &lt;dbl&gt; &lt;chr&gt;       &lt;dbl&gt;    &lt;dbl&gt; &lt;dbl&gt;   &lt;dbl&gt;
1 chr1   3069202  3069259 ENST00…     0 +         3069202  3069202     0       1
2 chr1   2412502  2412564 ENST00…     0 -         2412502  2412502     0       1
3 chr1  10474949 10474966 ENST00…     0 +        10474949 10474949     0       1
4 chr1   2476288  2476588 ENST00…     0 +         2476288  2476288     0       1
5 chr1   9292893  9355891 ENST00…     0 +         9292893  9292893     0       2
6 chr1   9069536  9069635 ENST00…     0 -         9069536  9069536     0       1
# ℹ 2 more variables: blockSizes &lt;dbl&gt;, blockStarts &lt;chr&gt;</code></pre>
</div>
</div>
<p>and intserct with the differential splicing results</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>diff.splicing.In<span class="fl">.5</span>UTR <span class="ot">&lt;-</span> diff.splicing.results <span class="sc">%&gt;%</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>    valr<span class="sc">::</span><span class="fu">bed_intersect</span>(., </span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>        fiveUTR,</span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">suffix=</span><span class="fu">c</span>(<span class="st">""</span>, <span class="st">".utr"</span>)</span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">FullyWithin =</span> start <span class="sc">&gt;=</span> start.utr <span class="sc">&amp;</span> end <span class="sc">&lt;=</span> end.utr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>And now count number significant differential</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>diff.splicing.In<span class="fl">.5</span>UTR <span class="sc">%&gt;%</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(IsSig) <span class="sc">%&gt;%</span></span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Contrast, cluster) <span class="sc">%&gt;%</span></span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">ContainsFullyWithin =</span> <span class="fu">any</span>(FullyWithin)) <span class="sc">%&gt;%</span></span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">distinct</span>(Contrast, cluster, <span class="at">.keep_all=</span>T) <span class="sc">%&gt;%</span></span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">count</span>(Contrast, ContainsFullyWithin) <span class="sc">%&gt;%</span></span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(Contrast) <span class="sc">%&gt;%</span></span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y=</span>Contrast, <span class="at">x=</span>n, <span class="at">fill=</span>ContainsFullyWithin)) <span class="sc">+</span></span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_col</span>(<span class="at">position=</span><span class="st">'stack'</span>) <span class="sc">+</span></span>
<span id="cb48-11"><a href="#cb48-11" aria-hidden="true" tabindex="-1"></a>    <span class="co"># geom_text(aes(label=n), hjust=-0.1) +</span></span>
<span id="cb48-12"><a href="#cb48-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_discrete</span>(<span class="at">limits =</span> rev) <span class="sc">+</span></span>
<span id="cb48-13"><a href="#cb48-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_continuous</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.1</span>))) <span class="sc">+</span></span>
<span id="cb48-14"><a href="#cb48-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb48-15"><a href="#cb48-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"Number of leafcutter significant clusters"</span>, <span class="at">y=</span><span class="st">"Contrast"</span>, <span class="at">caption=</span><span class="fu">str_glue</span>(<span class="st">"FDR&lt;1% &amp; |dPSI|&gt;{dPSI_threshold}%"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="20250716_Explore_DifferentialExpressionSplicingResults_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>I recall they are most interested in the 5UTR junctions that might be blocked by oligo to up-regulate expression. That is, the splice event that is down-regulated in the NMD enriched (or heavy polysome, highly translated) conditions, such that blocking the splice event would up-regulate the expression of the gene.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>diff.splicing.In<span class="fl">.5</span>UTR <span class="sc">%&gt;%</span></span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(IsSig) <span class="sc">%&gt;%</span></span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(deltapsi <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">10</span> <span class="sc">&amp;</span> FullyWithin) <span class="sc">%&gt;%</span></span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">distinct</span>(Contrast, Gene_name, <span class="at">.keep_all=</span>T) <span class="sc">%&gt;%</span></span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">count</span>(Contrast) <span class="sc">%&gt;%</span></span>
<span id="cb49-6"><a href="#cb49-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(Contrast) <span class="sc">%&gt;%</span></span>
<span id="cb49-7"><a href="#cb49-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y=</span>Contrast, <span class="at">x=</span>n)) <span class="sc">+</span></span>
<span id="cb49-8"><a href="#cb49-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_col</span>(<span class="at">position=</span><span class="st">'stack'</span>) <span class="sc">+</span></span>
<span id="cb49-9"><a href="#cb49-9" aria-hidden="true" tabindex="-1"></a>    <span class="co"># geom_text(aes(label=n), hjust=-0.1) +</span></span>
<span id="cb49-10"><a href="#cb49-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_discrete</span>(<span class="at">limits =</span> rev) <span class="sc">+</span></span>
<span id="cb49-11"><a href="#cb49-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_continuous</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.1</span>))) <span class="sc">+</span></span>
<span id="cb49-12"><a href="#cb49-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb49-13"><a href="#cb49-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"Number of differentially spliced genes"</span>, <span class="at">y=</span><span class="st">"Contrast"</span>, <span class="at">caption=</span><span class="fu">str_glue</span>(<span class="st">"FDR&lt;1% &amp; |dPSI|&gt;{dPSI_threshold}% &amp; deltaPSI &lt; -0.20 &amp; FullyWithin"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="20250716_Explore_DifferentialExpressionSplicingResults_files/figure-html/unnamed-chunk-21-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>One could probably do more here… Like look for overlap between some of the datasets, and which are disease relevant genes</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggVennDiagram)</span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a>venn_data <span class="ot">&lt;-</span> diff.splicing.In<span class="fl">.5</span>UTR <span class="sc">%&gt;%</span></span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="fu">str_detect</span>(Contrast, <span class="st">"Polysome"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(IsSig) <span class="sc">%&gt;%</span></span>
<span id="cb50-7"><a href="#cb50-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(deltapsi <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">10</span> <span class="sc">&amp;</span> FullyWithin) <span class="sc">%&gt;%</span></span>
<span id="cb50-8"><a href="#cb50-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">distinct</span>(Contrast, gene_names) <span class="sc">%&gt;%</span></span>
<span id="cb50-9"><a href="#cb50-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Contrast) <span class="sc">%&gt;%</span></span>
<span id="cb50-10"><a href="#cb50-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">genes =</span> <span class="fu">list</span>(<span class="fu">unique</span>(gene_names))) <span class="sc">%&gt;%</span></span>
<span id="cb50-11"><a href="#cb50-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">deframe</span>()</span>
<span id="cb50-12"><a href="#cb50-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-13"><a href="#cb50-13" aria-hidden="true" tabindex="-1"></a><span class="fu">ggVennDiagram</span>(venn_data) <span class="sc">+</span></span>
<span id="cb50-14"><a href="#cb50-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_gradient</span>(<span class="at">low =</span> <span class="st">"white"</span>, <span class="at">high =</span> <span class="st">"dodgerblue"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="20250716_Explore_DifferentialExpressionSplicingResults_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>And write out bed file for browsing…</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>diff.splicing.In<span class="fl">.5</span>UTR <span class="sc">%&gt;%</span></span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="fu">str_detect</span>(Contrast, <span class="st">"Polysome"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Contrast, cluster) <span class="sc">%&gt;%</span></span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="fu">any</span>(IsSig <span class="sc">&amp;</span> FullyWithin <span class="sc">&amp;</span> deltapsi <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">10</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">color =</span> scales<span class="sc">::</span><span class="fu">col_numeric</span>(</span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">palette =</span> scales<span class="sc">::</span><span class="fu">div_gradient_pal</span>(<span class="at">low =</span> <span class="st">"#4575b4"</span>, <span class="at">mid =</span> <span class="st">"#ffffbf"</span>, <span class="at">high =</span> <span class="st">"#d73027"</span>),</span>
<span id="cb51-7"><a href="#cb51-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">domain =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">50</span>, <span class="dv">50</span>)</span>
<span id="cb51-8"><a href="#cb51-8" aria-hidden="true" tabindex="-1"></a>    )(<span class="fu">pmax</span>(<span class="fu">pmin</span>(deltapsi, <span class="dv">50</span>), <span class="sc">-</span><span class="dv">50</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb51-9"><a href="#cb51-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">thickStart =</span> start, <span class="at">thickEnd =</span> end) <span class="sc">%&gt;%</span></span>
<span id="cb51-10"><a href="#cb51-10" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(chrom, start, end, cluster, deltapsi, strand, thickStart, thickEnd, color) <span class="sc">%&gt;%</span></span>
<span id="cb51-11"><a href="#cb51-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Contrast) <span class="sc">%&gt;%</span></span>
<span id="cb51-12"><a href="#cb51-12" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>Contrast) <span class="sc">%&gt;%</span></span>
<span id="cb51-13"><a href="#cb51-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_walk</span>(<span class="sc">~</span><span class="fu">write_tsv</span>(.x, <span class="fu">paste0</span>(<span class="st">"../code/scratch/Polysome_5UTR_splicing_"</span>, .y<span class="sc">$</span>Contrast[<span class="dv">1</span>], <span class="st">".bed"</span>), <span class="at">col_names =</span> <span class="cn">FALSE</span>))</span>
<span id="cb51-14"><a href="#cb51-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-15"><a href="#cb51-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-16"><a href="#cb51-16" aria-hidden="true" tabindex="-1"></a><span class="co"># python code/module_workflows/rna_seq/scripts/leafcutter2/scripts/Reformat_gtf.py  -i /project2/yangili1/bjf79/ReferenceGenomes/GRCh38_GencodeRelease44Comprehensive/Reference.gtf -o /dev/null -fa /project2/yangili1/bjf79/ReferenceGenomes/GRCh38_GencodeRelease44Comprehensive/Reference.fa -bed12_out code/scratch/Reference.FirstORF_AtLeast42aa.bed -infer_transcript_type_approach C -translation_approach E -infer_gene_type_approach B</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>From browsing, these bed files, here is an example in the COMMD5 gene. Across all three polysome profiling contrast, red junction isoform is less translated (less associated with heavy polysomes) compared to the blue junction. Note that the red junction isoform includes an upstream ORF, as noted by the red tick i highlighted above the gene structure, and the my custom re-formatting of the annotated transcript structures to denote the first ORF with thick bars.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="assets/COMMD5_Example.png" class="img-fluid figure-img"></p>
<figcaption>Example of COMMD5 gene splicing in polysome profiling contrasts</figcaption>
</figure>
</div>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




</body></html>